{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Functions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "On this lab we will put to practice some of the concepts we have learned on this past few days.\n",
    "\n",
    "`NOTE: On this lab you should try to write all the functions yourself using only the most basic of python syntax and without functions such as len, count, sum, max, min, in, etc. Give it a try. 🧑🏻‍💻👩🏻‍💻`\n",
    "\n",
    "The cell after each exercise contains a few tests to check if your function works as expected."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from mod.testing import *\n",
    "import unittest"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Write a function that returns the greater of two numbers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def greater(a,b):\n",
    "    \"\"\"\n",
    "    Esta función comprueba que valor es mayor de dos dados.\n",
    "    Args:\n",
    "        a, b -> valores a comparar\n",
    "    Return:\n",
    "        valor mayor -> si a > b --> a, sino b\n",
    "    \"\"\"\n",
    "    if a > b:\n",
    "        return a\n",
    "    else:\n",
    "        return b "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "....................................................................................................\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.097s\n",
      "\n",
      "OK\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_greater(greater)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Now write a function that returns the largest element on a list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def greatest(arr):\n",
    "    \"\"\"\n",
    "    Se trata de una función que devuelve el valor mayor de una lista.\n",
    "    Para ello, se asigna en primer lugar el primer valor de la lista, y se recorre esta para ver si existe otro valor mayor que el inicial.\n",
    "    De serlo, se le asigna a la variable, que luego será el return.\n",
    "    Arg:\n",
    "        arr -> lista \n",
    "    Return:\n",
    "        el valor mayor\n",
    "    \"\"\"\n",
    "    largest = arr[0]\n",
    "    for i in arr:\n",
    "        if i > largest:\n",
    "            largest = i\n",
    "    return largest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "....................................................................................................\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.101s\n",
      "\n",
      "OK\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_greatest(greatest)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Write a function that sums all the elements on a list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def sum_all(arr):\n",
    "    \"\"\"\n",
    "    Esta función lo que hace es iterar sobre los valores de una lista y los va sumando a una variable local para poder obterner el total.\n",
    "    arg:\n",
    "        lista con ints\n",
    "    return:\n",
    "        int con la suma de los valores de la lista\n",
    "    \"\"\"\n",
    "    num = 0\n",
    "    for i in arr:\n",
    "        num += i\n",
    "    return num"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "....................................................................................................\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.097s\n",
      "\n",
      "OK\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_sum(sum_all)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. Write another function that multiplies all the elements on a list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def mult_all(arr):\n",
    "    \"\"\"\n",
    "    Esta función lo que hace es iterar sobre los valores de una lista y los va multiplicando a partir de una variable local que se considera que es 1.\n",
    "    arg:\n",
    "        lista con ints\n",
    "    return:\n",
    "        int con la suma de los valores de la lista\n",
    "    \"\"\"\n",
    "    num = 1\n",
    "    for i in arr:\n",
    "        num *= i\n",
    "    return num"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "....................................................................................................\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.114s\n",
      "\n",
      "OK\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_mult(mult_all)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. Now combine those two ideas and write a function that receives a list and either \"+\" or \"*\" and outputs acordingly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "def oper_all(arr, oper):\n",
    "    \"\"\"\n",
    "    Esta función lo que hace hacer la operación indicada en una lista dada. Por tanto, permite elegir mediante la multiplicación de todos los elementos de una lista o la suma de todos los elementos de una lista.\n",
    "    arg:\n",
    "        lista con ints, operación a realizar\n",
    "    return:\n",
    "        int con la suma o la multiplicación de los valores de la lista\n",
    "    \"\"\"\n",
    "    if oper == '*':\n",
    "        num = 1\n",
    "        for i in arr:\n",
    "            num *= i\n",
    "        return num\n",
    "    elif oper == '+':\n",
    "        num = 0\n",
    "        for i in arr:\n",
    "            num += i\n",
    "        return num"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "....................................................................................................\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.111s\n",
      "\n",
      "OK\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_operations(oper_all)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 6. Write a function that returns the factorial of a number."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "def factorial(n):\n",
    "    \"\"\"\n",
    "    Esta función calcula el factorial de un número dado. Para ello, se multiplica un número n*(n-1)*...*3*2*1\n",
    "    para ello, se recorre un rango de largo n+1 y se va multiplicando mediante un contador todos los valores del rango.\n",
    "    arg:\n",
    "        int del que se hace el factorial\n",
    "    return:\n",
    "        int cpm el resultado de la operación\n",
    "    \"\"\"\n",
    "    count = 1\n",
    "    for i in range(1,n+1):\n",
    "        count = i * count\n",
    "    return count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "....................................................................................................\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.089s\n",
      "\n",
      "OK\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_factorial(factorial)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 7. Write a function that takes a list and returns a list of the unique values.\n",
    "\n",
    "`NOTE: You cannot use set. 🤔`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "def unique(arr):\n",
    "    \"\"\"\n",
    "    Esta función itera sobre una lista y comprueba si los items son o no únicos.\n",
    "    Si son únicos, se añaden a una lista que será el resultado.\n",
    "    arg:\n",
    "        arr como lista\n",
    "    return:\n",
    "        unique_list como lista de valores únicos\n",
    "    \"\"\"\n",
    "    unique_list = []\n",
    "    for i in arr:\n",
    "        if i in unique_list:\n",
    "            #print(i)\n",
    "            continue\n",
    "        else:\n",
    "            unique_list.append(i)    \n",
    "    return unique_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "....................................................................................................\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.276s\n",
      "\n",
      "OK\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_unique(unique)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 8. Write a function that returns the mode of a list, i.e.: the element that appears the most times.\n",
    "`NOTE: You should not use count... 🧐`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "##no sé que hago mal, pero no funciona\n",
    "def mode_counter(arr):\n",
    "    \"\"\"\n",
    "    En esta función se obtiene la moda sin usar los métodos determinados. La moda es el número que más aparece dentro de una distribución.\n",
    "    Para ello, se añade cada elemento de la lista como key, para poder forzar a que sean únicos y se les asigna el valor 1.\n",
    "    Si van apareciendo valores de clave repetidos, a la clave se le sumará 1.\n",
    "    Una vez montado el diccionario temporal, se generará la variable largest igual al primer valor y se itrerará por los valores del diccionario para ver si son mayores que el original, y si lo son sobreescribirlo\n",
    "    arg:\n",
    "        lista de la que se quiere calcular la moda\n",
    "    restul:\n",
    "        el valor de la moda\n",
    "    \"\"\"\n",
    "    dict_moda = {}\n",
    "    for i in arr:\n",
    "        dict_moda.setdefault(i, 0)\n",
    "        dict_moda[i] += 1\n",
    "    largest = list(dict_moda.values())[0]\n",
    "    for v in dict_moda.values():\n",
    "        #print(value, largest)\n",
    "        if v > largest:\n",
    "            largest = v\n",
    "        return largest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{1: 1, 2: 1, 3: 1, 4: 1, 5: 1, 6: 1, 7: 1, 8: 1, 9: 10} 1\n",
      "1 ---> 1\n",
      "Nope 1 ---> 1\n",
      "1 ---> 1\n",
      "Nope 1 ---> 1\n",
      "1 ---> 1\n",
      "Nope 1 ---> 1\n",
      "1 ---> 1\n",
      "Nope 1 ---> 1\n",
      "1 ---> 1\n",
      "Nope 1 ---> 1\n",
      "1 ---> 1\n",
      "Nope 1 ---> 1\n",
      "1 ---> 1\n",
      "Nope 1 ---> 1\n",
      "1 ---> 1\n",
      "Nope 1 ---> 1\n",
      "10 ---> 1\n",
      "Sip 10 ---> 10\n",
      "{1: 1, 2: 1, 3: 1, 4: 1, 5: 1, 6: 1, 7: 1, 8: 1, 9: 10} 10\n"
     ]
    }
   ],
   "source": [
    "arr = [1,2,3,4,5,6,7,8,9,9,9,9,9,9,9,9,9,9]\n",
    "dict_moda = {}\n",
    "for i in arr:\n",
    "    dict_moda.setdefault(i, 0)\n",
    "    dict_moda[i] += 1\n",
    "largest = list(dict_moda.values())[0]\n",
    "print(dict_moda, largest)\n",
    "for value in dict_moda.values():\n",
    "    print(value,'--->' ,largest)\n",
    "    if value > largest:\n",
    "        largest = value\n",
    "        print('Sip', value,'--->' ,largest)\n",
    "    else:\n",
    "        print('Nope', value,'--->' ,largest)\n",
    "        continue\n",
    "\n",
    "    print(dict_moda, largest)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "FFFFFFFFFFFFFFFFFFFFFFFFFF.FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 52 != 4 : Should be 4\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 20 : Should be 20\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 12 : Should be 12\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 8 : Should be 8\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 5 != 21 : Should be 21\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 8 != 19 : Should be 19\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 8 != 16 : Should be 16\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 22 : Should be 22\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 17 : Should be 17\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 9 : Should be 9\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 4 : Should be 4\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 5 != 1 : Should be 1\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 2 != 8 : Should be 8\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 13 : Should be 13\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 3 != 21 : Should be 21\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 18 : Should be 18\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 16 : Should be 16\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 1 != 20 : Should be 20\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 8 != 15 : Should be 15\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 8 != 23 : Should be 23\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 7 != 21 : Should be 21\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 12 : Should be 12\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 21 : Should be 21\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 15 : Should be 15\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 53 != 11 : Should be 11\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 3 != 25 : Should be 25\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 9 != 25 : Should be 25\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 5 != 23 : Should be 23\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 5 != 10 : Should be 10\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 3 != 25 : Should be 25\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 9 != 5 : Should be 5\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 3 != 6 : Should be 6\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 57 != 12 : Should be 12\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 3 != 22 : Should be 22\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 2 : Should be 2\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 20 : Should be 20\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 5 != 20 : Should be 20\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 2 : Should be 2\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 8 != 19 : Should be 19\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 5 != 6 : Should be 6\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 56 != 7 : Should be 7\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 5 != 9 : Should be 9\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 13 : Should be 13\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 5 != 21 : Should be 21\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 1 : Should be 1\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 22 : Should be 22\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 24 : Should be 24\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 7 != 11 : Should be 11\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 8 != 1 : Should be 1\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 8 : Should be 8\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 20 : Should be 20\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 19 : Should be 19\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 11 != 7 : Should be 7\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 3 != 14 : Should be 14\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 2 != 3 : Should be 3\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 5 != 8 : Should be 8\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 12 : Should be 12\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 10 != 23 : Should be 23\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 15 : Should be 15\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 8 : Should be 8\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 14 : Should be 14\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 17 : Should be 17\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 10 != 15 : Should be 15\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 3 != 14 : Should be 14\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 3 != 12 : Should be 12\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 19 : Should be 19\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 15 : Should be 15\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 4 : Should be 4\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 3 != 21 : Should be 21\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 8 : Should be 8\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 11 != 8 : Should be 8\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 13 : Should be 13\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 22 : Should be 22\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 7 != 19 : Should be 19\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 5 != 4 : Should be 4\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 8 != 21 : Should be 21\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 21 : Should be 21\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 1 : Should be 1\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 10 : Should be 10\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 11 : Should be 11\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 10 != 3 : Should be 3\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 8 != 1 : Should be 1\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 13 : Should be 13\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 7 != 8 : Should be 8\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 8 != 1 : Should be 1\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 24 : Should be 24\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 8 : Should be 8\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 9 : Should be 9\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 8 != 21 : Should be 21\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 5 != 3 : Should be 3\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 13 : Should be 13\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 5 != 11 : Should be 11\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 20 : Should be 20\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 3 != 7 : Should be 7\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 22 : Should be 22\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 5 != 6 : Should be 6\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 15 : Should be 15\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 4 != 10 : Should be 10\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_mode.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 127, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 6 != 5 : Should be 5\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.143s\n",
      "\n",
      "FAILED (failures=99)\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_mode(mode_counter)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 9. Write a function that calculates the standard deviation of a list.\n",
    "`NOTE: Do not use any libraries or already built functions. 😉`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "from statistics import stdev\n",
    "def st_dev(arr):\n",
    "    \"\"\"\n",
    "    Para calcular la desviación estándar primero hay que hacer la media de todos los elementos de la lista. \n",
    "    \\nPara hacer la media, se hace la suma de todos los elementos de la lista iterando y se divide entre el count de todos los elementos de la lista.\n",
    "    \\nPosteriormente, se calcula la varianza. Para ello se hace el sumatorio de cada valor de la lista menos la media de la lista elevado al cuadrado, y se divide entre la longitud de la lista -1 (stdev muestral)\n",
    "    \\nPara obtener la desviación estándar se hace finalmente le cuadrado de la varianza.\n",
    "    Arg:\n",
    "        lista\n",
    "    result:\n",
    "        desviación estándar\n",
    "    \"\"\"\n",
    "    sum = 0\n",
    "    count = 0\n",
    "    var = 0\n",
    "    variance = 0\n",
    "    for item in arr:\n",
    "        sum += item\n",
    "        count += 1\n",
    "    mean_ = sum / count\n",
    "    for item in arr:\n",
    "        var = var + ((item - mean_) ** 2)\n",
    "        variance = var / (count - 1)\n",
    "    st_dev = variance ** (1/2)\n",
    "    return st_dev"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "....................................................................................................\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.109s\n",
      "\n",
      "OK\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_stdev(st_dev)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 10. Write a function to check if a string is a pangram, i.e.: if it contains all the letters of the alphabet at least once. Mind that the strings may contain characters that are not letters."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "### no funciona la función\n",
    "def pangram(string):\n",
    "    \"\"\"\n",
    "    Esta función coge un string dado y lo compara con el abecedario, para ver si tiene todas las letras del abecedario.\n",
    "    \\nPrimero se generará un string con todos los caracteres que son estrictamente letras con el método .isalpha() que se añadirá a un string\n",
    "    \\nA continuación se iterará por el string con letras, y se contarán cada una de ellas. Si el contador fuera superior a 26, el número de letras del alfabeto debería devolver True.\n",
    "    args:\n",
    "        string\n",
    "    return:\n",
    "        True si tiene todas las letras del abecedario\n",
    "        False si le falta alguna letra\n",
    "    \"\"\"\n",
    "    count = 0\n",
    "    caracteres_alpha = ''\n",
    "    caracteres = 'abcdefghijklmnopqrstuvwxyz'\n",
    "    #print(string)\n",
    "    for i in str(string):\n",
    "        if i.isalpha():\n",
    "            caracteres_alpha += i\n",
    "    for l in str(caracteres_alpha):\n",
    "        if l.lower() in caracteres:\n",
    "            count += 1\n",
    "    #print(count, string)\n",
    "    if count >= 26:\n",
    "        return True\n",
    "    else:\n",
    "        return False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "...............FF.F.FFFFFFFFFF\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_pangram.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 172, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: True != False : Should be False\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_pangram.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 172, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: True != False : Should be False\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_pangram.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 172, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: True != False : Should be False\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_pangram.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 172, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: True != False : Should be False\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_pangram.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 172, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: True != False : Should be False\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_pangram.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 172, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: True != False : Should be False\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_pangram.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 172, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: True != False : Should be False\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_pangram.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 172, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: True != False : Should be False\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_pangram.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 172, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: True != False : Should be False\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_pangram.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 172, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: True != False : Should be False\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_pangram.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 172, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: True != False : Should be False\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_pangram.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 172, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: True != False : Should be False\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_pangram.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 172, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: True != False : Should be False\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "Ran 30 tests in 0.039s\n",
      "\n",
      "FAILED (failures=13)\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_pangram(pangram)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 11. Write a function that receives a string of comma separated words and returns a string of comma separated words sorted alphabetically.\n",
    "\n",
    "`NOTE: You may use sorted but not split and definitely no join! 🤪`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "## no sé que está mal en la función, pero el test da error...\n",
    "## comparando los resultadosestán igual, no sé si se está realizando alguna opeación no permitida\n",
    "def sort_alpha(string):\n",
    "    \"\"\"\n",
    "    Esta función toma un caracter con comas y busca ordenar cada uno de los trozos separados por comas alfabéticamente.\n",
    "    Primero, recorre los caracteres del string y genera una lista con cada palabra separada por comas.\n",
    "    Se ordena esa lista alfabéticamente y se appendea a una nueva palabra.\n",
    "    arg:\n",
    "        un string con comas\n",
    "    return:\n",
    "        string ordenado alfabeticamente y separado por comas\n",
    "    \"\"\"\n",
    "    lista_palabras = []\n",
    "    palabra = ''\n",
    "    palabra_final = ''\n",
    "    for c in str(string):\n",
    "        palabra += c\n",
    "        if c == ',':\n",
    "            lista_palabras.append(palabra)\n",
    "            palabra = ''\n",
    "    for i in sorted(lista_palabras):\n",
    "        palabra_final += i\n",
    "    #print(string, lista_palabras, palabra_final[:-1])\n",
    "    return palabra_final[:-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bieuts,gaisrers,ifkpx,pkkhxbkeni,rqtihtldz,sayqod,tjis,vmwt' != 'bieuts,drtnyvnqkh,gaisrers,ifkpx,pkkhxbkeni,rqtihtldz,sayqod,tjis,vmwt'\n",
      "- bieuts,gaisrers,ifkpx,pkkhxbkeni,rqtihtldz,sayqod,tjis,vmwt\n",
      "+ bieuts,drtnyvnqkh,gaisrers,ifkpx,pkkhxbkeni,rqtihtldz,sayqod,tjis,vmwt\n",
      "?       +++++++++++\n",
      " : Should be bieuts,drtnyvnqkh,gaisrers,ifkpx,pkkhxbkeni,rqtihtldz,sayqod,tjis,vmwt\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'anxi[43 chars]btol,mige,poiis,rvhmbwhyg,ryqiflv,snmoi,sqfttd[23 chars]yzxa' != 'anxi[43 chars]btol,ivbgcljqlq,mige,poiis,rvhmbwhyg,ryqiflv,s[34 chars]yzxa'\n",
      "- anxia,aqcjbrlwd,arufhyk,bzgil,glgw,gswod,gywqdrbtol,mige,poiis,rvhmbwhyg,ryqiflv,snmoi,sqfttdvw,uemxbvkt,xlregk,yqnfyzxa\n",
      "+ anxia,aqcjbrlwd,arufhyk,bzgil,glgw,gswod,gywqdrbtol,ivbgcljqlq,mige,poiis,rvhmbwhyg,ryqiflv,snmoi,sqfttdvw,uemxbvkt,xlregk,yqnfyzxa\n",
      "?                                                    +++++++++++\n",
      " : Should be anxia,aqcjbrlwd,arufhyk,bzgil,glgw,gswod,gywqdrbtol,ivbgcljqlq,mige,poiis,rvhmbwhyg,ryqiflv,snmoi,sqfttdvw,uemxbvkt,xlregk,yqnfyzxa\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'ajze[62 chars]vw,jjknbrj,msflxiwjaz,uxqggcpi,yyavlhaps,zjqhjf,zvlmpmurh' != 'ajze[62 chars]vw,jjknbrj,msflxiwjaz,uxqggcpi,yyavlhaps,zeyq,zjqhjf,zvlmpmurh'\n",
      "- ajzevnn,anvjxio,asgtn,axxvexp,bvrmmpp,eafyooswdq,ejskpvrkzd,guizfovw,jjknbrj,msflxiwjaz,uxqggcpi,yyavlhaps,zjqhjf,zvlmpmurh\n",
      "+ ajzevnn,anvjxio,asgtn,axxvexp,bvrmmpp,eafyooswdq,ejskpvrkzd,guizfovw,jjknbrj,msflxiwjaz,uxqggcpi,yyavlhaps,zeyq,zjqhjf,zvlmpmurh\n",
      "?                                                                                                             +++++\n",
      " : Should be ajzevnn,anvjxio,asgtn,axxvexp,bvrmmpp,eafyooswdq,ejskpvrkzd,guizfovw,jjknbrj,msflxiwjaz,uxqggcpi,yyavlhaps,zeyq,zjqhjf,zvlmpmurh\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bxjpwy,cqchu,dxjddkjjr,hcpsr,jylziemwt,lmjx,ombtnzk,rqgkxym,wfugfz' != 'bxjpwy,cqchu,dxjddkjjr,hcpsr,jylziemwt,lmjx,ombtnzk,rqgkxym,tasrltn,wfugfz'\n",
      "- bxjpwy,cqchu,dxjddkjjr,hcpsr,jylziemwt,lmjx,ombtnzk,rqgkxym,wfugfz\n",
      "+ bxjpwy,cqchu,dxjddkjjr,hcpsr,jylziemwt,lmjx,ombtnzk,rqgkxym,tasrltn,wfugfz\n",
      "?                                                             ++++++++\n",
      " : Should be bxjpwy,cqchu,dxjddkjjr,hcpsr,jylziemwt,lmjx,ombtnzk,rqgkxym,tasrltn,wfugfz\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bsvb[21 chars],hheg,ldtmvvjfwe,lfbjya,nazlw,nyyrpxfw,sjpusgwm,wbtzckl' != 'bsvb[21 chars],hheg,itcpxq,ldtmvvjfwe,lfbjya,nazlw,nyyrpxfw,sjpusgwm,wbtzckl'\n",
      "- bsvbt,faltjyzgi,fcbernwvr,hheg,ldtmvvjfwe,lfbjya,nazlw,nyyrpxfw,sjpusgwm,wbtzckl\n",
      "+ bsvbt,faltjyzgi,fcbernwvr,hheg,itcpxq,ldtmvvjfwe,lfbjya,nazlw,nyyrpxfw,sjpusgwm,wbtzckl\n",
      "?                               +++++++\n",
      " : Should be bsvbt,faltjyzgi,fcbernwvr,hheg,itcpxq,ldtmvvjfwe,lfbjya,nazlw,nyyrpxfw,sjpusgwm,wbtzckl\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'cxflchm,gieztkfkj,hscg,irzazoy,ofnrzmnuft,pfzwvmxz,pokpizor,[35 chars]nfpp' != 'cxflchm,gieztkfkj,hikwpbfzp,hscg,irzazoy,ofnrzmnuft,pfzwvmxz[45 chars]nfpp'\n",
      "- cxflchm,gieztkfkj,hscg,irzazoy,ofnrzmnuft,pfzwvmxz,pokpizor,rwjhe,sgjjd,tclvbymtqm,wgsjq,xyubzynfpp\n",
      "+ cxflchm,gieztkfkj,hikwpbfzp,hscg,irzazoy,ofnrzmnuft,pfzwvmxz,pokpizor,rwjhe,sgjjd,tclvbymtqm,wgsjq,xyubzynfpp\n",
      "?                  ++++++++++\n",
      " : Should be cxflchm,gieztkfkj,hikwpbfzp,hscg,irzazoy,ofnrzmnuft,pfzwvmxz,pokpizor,rwjhe,sgjjd,tclvbymtqm,wgsjq,xyubzynfpp\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bloq[140 chars]omdj,viucxi,vqpvi,wgyax,xshfn,yqqzzshv,ytqp,zpcfywcd' != 'bloq[140 chars]omdj,ttuwhdsyas,viucxi,vqpvi,wgyax,xshfn,yqqzz[13 chars]ywcd'\n",
      "- bloq,bpgeq,eqdcyilho,giedbxcz,gnmh,hwdquosx,hxsdnfw,idiyryw,iijgxhrcc,jdstvualvi,jihmmsjgy,ldvvvzvx,litdoebrs,loyqpweza,oxabkavmnj,qpgcjoko,soueomdj,viucxi,vqpvi,wgyax,xshfn,yqqzzshv,ytqp,zpcfywcd\n",
      "+ bloq,bpgeq,eqdcyilho,giedbxcz,gnmh,hwdquosx,hxsdnfw,idiyryw,iijgxhrcc,jdstvualvi,jihmmsjgy,ldvvvzvx,litdoebrs,loyqpweza,oxabkavmnj,qpgcjoko,soueomdj,ttuwhdsyas,viucxi,vqpvi,wgyax,xshfn,yqqzzshv,ytqp,zpcfywcd\n",
      "?                                                                                                                                                      +++++++++++\n",
      " : Should be bloq,bpgeq,eqdcyilho,giedbxcz,gnmh,hwdquosx,hxsdnfw,idiyryw,iijgxhrcc,jdstvualvi,jihmmsjgy,ldvvvzvx,litdoebrs,loyqpweza,oxabkavmnj,qpgcjoko,soueomdj,ttuwhdsyas,viucxi,vqpvi,wgyax,xshfn,yqqzzshv,ytqp,zpcfywcd\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bbeojaij,kajzq,ntlzhpgltf,pyftqek,wcor,xwvqx' != 'bbeojaij,dczlw,kajzq,ntlzhpgltf,pyftqek,wcor,xwvqx'\n",
      "- bbeojaij,kajzq,ntlzhpgltf,pyftqek,wcor,xwvqx\n",
      "+ bbeojaij,dczlw,kajzq,ntlzhpgltf,pyftqek,wcor,xwvqx\n",
      "?         ++++++\n",
      " : Should be bbeojaij,dczlw,kajzq,ntlzhpgltf,pyftqek,wcor,xwvqx\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'aixohl,boprh,coisz,pxjhg,rpia,soup,tqzllqvkf' != 'aixohl,boprh,coisz,pxjhg,rpia,soup,tqzllqvkf,vrpdmo'\n",
      "- aixohl,boprh,coisz,pxjhg,rpia,soup,tqzllqvkf\n",
      "+ aixohl,boprh,coisz,pxjhg,rpia,soup,tqzllqvkf,vrpdmo\n",
      "?                                             +++++++\n",
      " : Should be aixohl,boprh,coisz,pxjhg,rpia,soup,tqzllqvkf,vrpdmo\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'arah[94 chars]wfke,oedvibrhm,oinvox,pbivku,pfpwi,sybfgbj,vsm[28 chars]wqap' != 'arah[94 chars]wfke,mliveqbgd,oedvibrhm,oinvox,pbivku,pfpwi,s[38 chars]wqap'\n",
      "- arahbm,bgbmzgmluf,crrh,drlh,ejndq,gtde,gzijo,hojbjgcy,hovgeb,jkwjgflnj,jsivdhwn,kazrycaef,kigx,mkzwfke,oedvibrhm,oinvox,pbivku,pfpwi,sybfgbj,vsmzcualtu,wdueyhmpag,wqakkocf,wqap\n",
      "+ arahbm,bgbmzgmluf,crrh,drlh,ejndq,gtde,gzijo,hojbjgcy,hovgeb,jkwjgflnj,jsivdhwn,kazrycaef,kigx,mkzwfke,mliveqbgd,oedvibrhm,oinvox,pbivku,pfpwi,sybfgbj,vsmzcualtu,wdueyhmpag,wqakkocf,wqap\n",
      "?                                                                                                        ++++++++++\n",
      " : Should be arahbm,bgbmzgmluf,crrh,drlh,ejndq,gtde,gzijo,hojbjgcy,hovgeb,jkwjgflnj,jsivdhwn,kazrycaef,kigx,mkzwfke,mliveqbgd,oedvibrhm,oinvox,pbivku,pfpwi,sybfgbj,vsmzcualtu,wdueyhmpag,wqakkocf,wqap\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'euosa,fajbe,foxpf,gabbmqo,ibarov,jyprkqh,[111 chars]menu' != 'dagrsdn,euosa,fajbe,foxpf,gabbmqo,ibarov,[119 chars]menu'\n",
      "- euosa,fajbe,foxpf,gabbmqo,ibarov,jyprkqh,kgzkihwgb,kujc,lgatui,luiuor,mulsom,nvkzlwv,ovocgwqkf,phffeiip,preu,uoyxfm,uxilsncrz,vxxbj,xfdofwcqqd,xvfsm,zimmenu\n",
      "+ dagrsdn,euosa,fajbe,foxpf,gabbmqo,ibarov,jyprkqh,kgzkihwgb,kujc,lgatui,luiuor,mulsom,nvkzlwv,ovocgwqkf,phffeiip,preu,uoyxfm,uxilsncrz,vxxbj,xfdofwcqqd,xvfsm,zimmenu\n",
      "? ++++++++\n",
      " : Should be dagrsdn,euosa,fajbe,foxpf,gabbmqo,ibarov,jyprkqh,kgzkihwgb,kujc,lgatui,luiuor,mulsom,nvkzlwv,ovocgwqkf,phffeiip,preu,uoyxfm,uxilsncrz,vxxbj,xfdofwcqqd,xvfsm,zimmenu\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bczfqpwvzv,fmnms,gpgrkzy,gpiallj,gsyo,hfj[113 chars]onmf' != 'aoawjae,bczfqpwvzv,fmnms,gpgrkzy,gpiallj,[121 chars]onmf'\n",
      "- bczfqpwvzv,fmnms,gpgrkzy,gpiallj,gsyo,hfjaeuh,hlubrs,ihfrcxqxf,jnrjtz,jqqhobjwqu,kgxuvgqj,kkzzcoo,krwlq,ldbhtwpn,luaghe,mlsniaycm,ngpk,nqvsgat,nrau,pvshxeonmf\n",
      "+ aoawjae,bczfqpwvzv,fmnms,gpgrkzy,gpiallj,gsyo,hfjaeuh,hlubrs,ihfrcxqxf,jnrjtz,jqqhobjwqu,kgxuvgqj,kkzzcoo,krwlq,ldbhtwpn,luaghe,mlsniaycm,ngpk,nqvsgat,nrau,pvshxeonmf\n",
      "? ++++++++\n",
      " : Should be aoawjae,bczfqpwvzv,fmnms,gpgrkzy,gpiallj,gsyo,hfjaeuh,hlubrs,ihfrcxqxf,jnrjtz,jqqhobjwqu,kgxuvgqj,kkzzcoo,krwlq,ldbhtwpn,luaghe,mlsniaycm,ngpk,nqvsgat,nrau,pvshxeonmf\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'afpnwuob,dzvdbuuqgi,jzxytydme,npcv,tpmysie' != 'afpnwuob,dzvdbuuqgi,jzxytydme,npcv,tpmysie,wnrklukyc'\n",
      "- afpnwuob,dzvdbuuqgi,jzxytydme,npcv,tpmysie\n",
      "+ afpnwuob,dzvdbuuqgi,jzxytydme,npcv,tpmysie,wnrklukyc\n",
      "?                                           ++++++++++\n",
      " : Should be afpnwuob,dzvdbuuqgi,jzxytydme,npcv,tpmysie,wnrklukyc\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'cedznjjmnq,expwkspe,foheqrnft,ntnhiuog,siog,vcunpzdxkc,xhlsh' != 'cedznjjmnq,expwkspe,foheqrnft,ntnhiuog,siog,trtfonr,vcunpzdxkc,xhlsh'\n",
      "- cedznjjmnq,expwkspe,foheqrnft,ntnhiuog,siog,vcunpzdxkc,xhlsh\n",
      "+ cedznjjmnq,expwkspe,foheqrnft,ntnhiuog,siog,trtfonr,vcunpzdxkc,xhlsh\n",
      "?                                             ++++++++\n",
      " : Should be cedznjjmnq,expwkspe,foheqrnft,ntnhiuog,siog,trtfonr,vcunpzdxkc,xhlsh\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'aitmiubglh,dhzgxt,gnis,hnhpc,jokx,oicwbrfy,qojxhor,snfkc,zjnpxku' != 'aitmiubglh,dhzgxt,gnis,hnhpc,jokx,oicwbrfy,qojxhor,rihr,snfkc,zjnpxku'\n",
      "- aitmiubglh,dhzgxt,gnis,hnhpc,jokx,oicwbrfy,qojxhor,snfkc,zjnpxku\n",
      "+ aitmiubglh,dhzgxt,gnis,hnhpc,jokx,oicwbrfy,qojxhor,rihr,snfkc,zjnpxku\n",
      "?                                                    +++++\n",
      " : Should be aitmiubglh,dhzgxt,gnis,hnhpc,jokx,oicwbrfy,qojxhor,rihr,snfkc,zjnpxku\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'cepvuj,dtqkaps,flqowyk,htebvmgqlt' != 'cepvuj,dtqkaps,flqowyk,htebvmgqlt,obcbdsy'\n",
      "- cepvuj,dtqkaps,flqowyk,htebvmgqlt\n",
      "+ cepvuj,dtqkaps,flqowyk,htebvmgqlt,obcbdsy\n",
      "?                                  ++++++++\n",
      " : Should be cepvuj,dtqkaps,flqowyk,htebvmgqlt,obcbdsy\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'hphffk,mtii,nahzm,orjrey,pwkzfxtus,urudu,wegoht' != 'hphffk,mtii,nahzm,orjrey,oslyxdzntq,pwkzfxtus,urudu,wegoht'\n",
      "- hphffk,mtii,nahzm,orjrey,pwkzfxtus,urudu,wegoht\n",
      "+ hphffk,mtii,nahzm,orjrey,oslyxdzntq,pwkzfxtus,urudu,wegoht\n",
      "?                          +++++++++++\n",
      " : Should be hphffk,mtii,nahzm,orjrey,oslyxdzntq,pwkzfxtus,urudu,wegoht\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'cloh,fadefqrvyq,gupfsef,npwi,nuhwryzrt,nvfc,oi[59 chars]udcy' != 'cloh,cyesjwysdi,fadefqrvyq,gupfsef,npwi,nuhwry[70 chars]udcy'\n",
      "- cloh,fadefqrvyq,gupfsef,npwi,nuhwryzrt,nvfc,oibg,ozxwyckeee,tietbgwxlo,ttahpgbhq,ujjnhvjsnf,uunbusanj,yvxudcy\n",
      "+ cloh,cyesjwysdi,fadefqrvyq,gupfsef,npwi,nuhwryzrt,nvfc,oibg,ozxwyckeee,tietbgwxlo,ttahpgbhq,ujjnhvjsnf,uunbusanj,yvxudcy\n",
      "?     +++++++++++\n",
      " : Should be cloh,cyesjwysdi,fadefqrvyq,gupfsef,npwi,nuhwryzrt,nvfc,oibg,ozxwyckeee,tietbgwxlo,ttahpgbhq,ujjnhvjsnf,uunbusanj,yvxudcy\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'azoe[13 chars]kty,oehleioumx,oztkvpnn,qkyzdudw,vmvzjkvu,xxvfrgr,zdmqas' != 'azoe[13 chars]kty,oehleioumx,oztkvpnn,qkyzdudw,vmvzjkvu,woolc,xxvfrgr,zdmqas'\n",
      "- azoe,cpar,gdxkudkkty,oehleioumx,oztkvpnn,qkyzdudw,vmvzjkvu,xxvfrgr,zdmqas\n",
      "+ azoe,cpar,gdxkudkkty,oehleioumx,oztkvpnn,qkyzdudw,vmvzjkvu,woolc,xxvfrgr,zdmqas\n",
      "?                                                            ++++++\n",
      " : Should be azoe,cpar,gdxkudkkty,oehleioumx,oztkvpnn,qkyzdudw,vmvzjkvu,woolc,xxvfrgr,zdmqas\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'afaq[43 chars],glkcjqme,ixcjl,lqicmbopb,peltmjqs,rtulavy,vkqfsla,xbsbzb' != 'afaq[43 chars],glkcjqme,ixcjl,lqicmbopb,peltmjqs,rtulavy,vkqfsla,xbsbzb,zqjz'\n",
      "- afaqnby,dffi,dhrlfqqwra,dijo,fmemhohe,gjzvoeysu,glkcjqme,ixcjl,lqicmbopb,peltmjqs,rtulavy,vkqfsla,xbsbzb\n",
      "+ afaqnby,dffi,dhrlfqqwra,dijo,fmemhohe,gjzvoeysu,glkcjqme,ixcjl,lqicmbopb,peltmjqs,rtulavy,vkqfsla,xbsbzb,zqjz\n",
      "?                                                                                                         +++++\n",
      " : Should be afaqnby,dffi,dhrlfqqwra,dijo,fmemhohe,gjzvoeysu,glkcjqme,ixcjl,lqicmbopb,peltmjqs,rtulavy,vkqfsla,xbsbzb,zqjz\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'fjejbecpwl,hiitfwgtxx,ktplh,kzfp,tbgtwnn,[28 chars]swkt' != 'czeyk,fjejbecpwl,hiitfwgtxx,ktplh,kzfp,tb[34 chars]swkt'\n",
      "- fjejbecpwl,hiitfwgtxx,ktplh,kzfp,tbgtwnn,tnpl,tukimotmc,txckosgijn,zsswkt\n",
      "+ czeyk,fjejbecpwl,hiitfwgtxx,ktplh,kzfp,tbgtwnn,tnpl,tukimotmc,txckosgijn,zsswkt\n",
      "? ++++++\n",
      " : Should be czeyk,fjejbecpwl,hiitfwgtxx,ktplh,kzfp,tbgtwnn,tnpl,tukimotmc,txckosgijn,zsswkt\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'csgg[54 chars]asaq,trqorvua,uqsfyorg,utlfbiq,wrlb,xwtlwgik,ybdyohwsl' != 'csgg[54 chars]asaq,trqorvua,uqsfyorg,utlfbiq,wrlb,xwtlwgik,ybdyohwsl,zduqewb'\n",
      "- csggmqvn,dtky,fbzurwcoqz,lnjpb,oabgdbigse,pgffovhy,qgehc,tasaq,trqorvua,uqsfyorg,utlfbiq,wrlb,xwtlwgik,ybdyohwsl\n",
      "+ csggmqvn,dtky,fbzurwcoqz,lnjpb,oabgdbigse,pgffovhy,qgehc,tasaq,trqorvua,uqsfyorg,utlfbiq,wrlb,xwtlwgik,ybdyohwsl,zduqewb\n",
      "?                                                                                                                 ++++++++\n",
      " : Should be csggmqvn,dtky,fbzurwcoqz,lnjpb,oabgdbigse,pgffovhy,qgehc,tasaq,trqorvua,uqsfyorg,utlfbiq,wrlb,xwtlwgik,ybdyohwsl,zduqewb\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'covlvqdp,fobt,gajygotflg' != 'covlvqdp,fobt,gajygotflg,vopakjc'\n",
      "- covlvqdp,fobt,gajygotflg\n",
      "+ covlvqdp,fobt,gajygotflg,vopakjc\n",
      "?                         ++++++++\n",
      " : Should be covlvqdp,fobt,gajygotflg,vopakjc\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'dvrd[27 chars]uch,jldvukrjnj,lbecvohxhs,lokazyb,mblekhb,ndor[49 chars]tspo' != 'dvrd[27 chars]uch,jkexd,jldvukrjnj,lbecvohxhs,lokazyb,mblekh[55 chars]tspo'\n",
      "- dvrd,gpktd,hlhjltr,hpltq,iuklmmuch,jldvukrjnj,lbecvohxhs,lokazyb,mblekhb,ndor,npmb,pkwfqwmha,sldwyxl,ulelhteyu,uuyinou,wvlsq,ytspo\n",
      "+ dvrd,gpktd,hlhjltr,hpltq,iuklmmuch,jkexd,jldvukrjnj,lbecvohxhs,lokazyb,mblekhb,ndor,npmb,pkwfqwmha,sldwyxl,ulelhteyu,uuyinou,wvlsq,ytspo\n",
      "?                                   ++++++\n",
      " : Should be dvrd,gpktd,hlhjltr,hpltq,iuklmmuch,jkexd,jldvukrjnj,lbecvohxhs,lokazyb,mblekhb,ndor,npmb,pkwfqwmha,sldwyxl,ulelhteyu,uuyinou,wvlsq,ytspo\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'ajqg[34 chars]inf,olnfhgy,qhbaxxa,rlsnau,siqovj,vmor,xhhivgoc,zuas' != 'ajqg[34 chars]inf,olnfhgy,qhbaxxa,rlsnau,siqovj,vmor,vmpihifal,xhhivgoc,zuas'\n",
      "- ajqgbdzycu,cjawmvo,eotcaqxgfk,epvzc,ezinf,olnfhgy,qhbaxxa,rlsnau,siqovj,vmor,xhhivgoc,zuas\n",
      "+ ajqgbdzycu,cjawmvo,eotcaqxgfk,epvzc,ezinf,olnfhgy,qhbaxxa,rlsnau,siqovj,vmor,vmpihifal,xhhivgoc,zuas\n",
      "?                                                                              ++++++++++\n",
      " : Should be ajqgbdzycu,cjawmvo,eotcaqxgfk,epvzc,ezinf,olnfhgy,qhbaxxa,rlsnau,siqovj,vmor,vmpihifal,xhhivgoc,zuas\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'cxdc[21 chars]jlrzh,fedqfhojy,iiili,iumvwb,jdpdi,mxyuz,wzupcbim,yyjid' != 'cxdc[21 chars]jlrzh,fedqfhojy,iiili,iumvwb,jdpdi,mxyuz,vfebgq,wzupcbim,yyjid'\n",
      "- cxdcscmvf,dphbasbhrx,dposjlrzh,fedqfhojy,iiili,iumvwb,jdpdi,mxyuz,wzupcbim,yyjid\n",
      "+ cxdcscmvf,dphbasbhrx,dposjlrzh,fedqfhojy,iiili,iumvwb,jdpdi,mxyuz,vfebgq,wzupcbim,yyjid\n",
      "?                                                                   +++++++\n",
      " : Should be cxdcscmvf,dphbasbhrx,dposjlrzh,fedqfhojy,iiili,iumvwb,jdpdi,mxyuz,vfebgq,wzupcbim,yyjid\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'akgg[67 chars]gmjn,pdfvdptq,pxgdyoruvi,rhjnmckl,xjyjhngkyz,yboghxk' != 'akgg[67 chars]gmjn,pdfvdptq,pwybtwrcz,pxgdyoruvi,rhjnmckl,xjyjhngkyz,yboghxk'\n",
      "- akggydb,awixee,bghbhakqhv,gcct,hvjxytmkb,icfyoo,itamowdx,jdcehdo,lvvt,ngmjn,pdfvdptq,pxgdyoruvi,rhjnmckl,xjyjhngkyz,yboghxk\n",
      "+ akggydb,awixee,bghbhakqhv,gcct,hvjxytmkb,icfyoo,itamowdx,jdcehdo,lvvt,ngmjn,pdfvdptq,pwybtwrcz,pxgdyoruvi,rhjnmckl,xjyjhngkyz,yboghxk\n",
      "?                                                                                       ++++++++++\n",
      " : Should be akggydb,awixee,bghbhakqhv,gcct,hvjxytmkb,icfyoo,itamowdx,jdcehdo,lvvt,ngmjn,pdfvdptq,pwybtwrcz,pxgdyoruvi,rhjnmckl,xjyjhngkyz,yboghxk\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'hnrhu,jkuj,jutum,qcvtxebth,sgpodklwez,vvowf' != 'hnrhu,jkuj,jutum,kheirly,qcvtxebth,sgpodklwez,vvowf'\n",
      "- hnrhu,jkuj,jutum,qcvtxebth,sgpodklwez,vvowf\n",
      "+ hnrhu,jkuj,jutum,kheirly,qcvtxebth,sgpodklwez,vvowf\n",
      "?                 ++++++++\n",
      " : Should be hnrhu,jkuj,jutum,kheirly,qcvtxebth,sgpodklwez,vvowf\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'atbl[29 chars]vkkv,feld,gveaj,jvkvybkfbj,kmwb,lppvrgvueb,mkd[13 chars]lwis' != 'atbl[29 chars]vkkv,emffsl,feld,gveaj,jvkvybkfbj,kmwb,lppvrgv[20 chars]lwis'\n",
      "- atblisfl,atugs,cfqhkhevy,ekyvt,elvkkv,feld,gveaj,jvkvybkfbj,kmwb,lppvrgvueb,mkdu,ndmypu,zzvxlwis\n",
      "+ atblisfl,atugs,cfqhkhevy,ekyvt,elvkkv,emffsl,feld,gveaj,jvkvybkfbj,kmwb,lppvrgvueb,mkdu,ndmypu,zzvxlwis\n",
      "?                                      +++++++\n",
      " : Should be atblisfl,atugs,cfqhkhevy,ekyvt,elvkkv,emffsl,feld,gveaj,jvkvybkfbj,kmwb,lppvrgvueb,mkdu,ndmypu,zzvxlwis\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bmdt[63 chars]hzr,msxwjetot,nfxuiay,nypl,odjebsgcwf,odlgwi,o[76 chars]vpas' != 'bmdt[63 chars]hzr,mgdzbbgcq,msxwjetot,nfxuiay,nypl,odjebsgcw[86 chars]vpas'\n",
      "- bmdtednm,cwcrhwgsqh,ddwqpkjbzx,ensjgwl,ewiaegym,fyre,kxrsqpufee,lathzr,msxwjetot,nfxuiay,nypl,odjebsgcwf,odlgwi,oqxxl,orhx,oxairlmwkg,ryzbk,sddsdfa,tavpm,tkegei,uiadgquljh,wghcj,yrxziidli,yvpas\n",
      "+ bmdtednm,cwcrhwgsqh,ddwqpkjbzx,ensjgwl,ewiaegym,fyre,kxrsqpufee,lathzr,mgdzbbgcq,msxwjetot,nfxuiay,nypl,odjebsgcwf,odlgwi,oqxxl,orhx,oxairlmwkg,ryzbk,sddsdfa,tavpm,tkegei,uiadgquljh,wghcj,yrxziidli,yvpas\n",
      "?                                                                       ++++++++++\n",
      " : Should be bmdtednm,cwcrhwgsqh,ddwqpkjbzx,ensjgwl,ewiaegym,fyre,kxrsqpufee,lathzr,mgdzbbgcq,msxwjetot,nfxuiay,nypl,odjebsgcwf,odlgwi,oqxxl,orhx,oxairlmwkg,ryzbk,sddsdfa,tavpm,tkegei,uiadgquljh,wghcj,yrxziidli,yvpas\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'ihrkdu,oxvbtrne,rklscr' != 'ihrkdu,nzqc,oxvbtrne,rklscr'\n",
      "- ihrkdu,oxvbtrne,rklscr\n",
      "+ ihrkdu,nzqc,oxvbtrne,rklscr\n",
      "?       +++++\n",
      " : Should be ihrkdu,nzqc,oxvbtrne,rklscr\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'fdqjmbl,fizn,kzyvqkqxeg,oflj,ogblt,orhoa,[40 chars]zdzl' != 'dtehgd,fdqjmbl,fizn,kzyvqkqxeg,oflj,ogblt[47 chars]zdzl'\n",
      "- fdqjmbl,fizn,kzyvqkqxeg,oflj,ogblt,orhoa,rlftjcjm,sdkksyzq,umsiourxp,uqgoctkhjf,vzdzl\n",
      "+ dtehgd,fdqjmbl,fizn,kzyvqkqxeg,oflj,ogblt,orhoa,rlftjcjm,sdkksyzq,umsiourxp,uqgoctkhjf,vzdzl\n",
      "? +++++++\n",
      " : Should be dtehgd,fdqjmbl,fizn,kzyvqkqxeg,oflj,ogblt,orhoa,rlftjcjm,sdkksyzq,umsiourxp,uqgoctkhjf,vzdzl\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'ayzjcmv,cdztuuwhr,gxtddq,sksipfmel,uvltbbf,vyhf,yafgp' != 'ayzjcmv,cdztuuwhr,ephemvsg,gxtddq,sksipfmel,uvltbbf,vyhf,yafgp'\n",
      "- ayzjcmv,cdztuuwhr,gxtddq,sksipfmel,uvltbbf,vyhf,yafgp\n",
      "+ ayzjcmv,cdztuuwhr,ephemvsg,gxtddq,sksipfmel,uvltbbf,vyhf,yafgp\n",
      "?                  +++++++++\n",
      " : Should be ayzjcmv,cdztuuwhr,ephemvsg,gxtddq,sksipfmel,uvltbbf,vyhf,yafgp\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'kffuicd,lqridwd,recf,xluvzgw' != 'kffuicd,lqridwd,recf,xluvzgw,yttw'\n",
      "- kffuicd,lqridwd,recf,xluvzgw\n",
      "+ kffuicd,lqridwd,recf,xluvzgw,yttw\n",
      "?                             +++++\n",
      " : Should be kffuicd,lqridwd,recf,xluvzgw,yttw\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bklx[112 chars]pwmw,sllbie,trieh,wetox,wntnomxeyv,wuolruphy,yjfpj,zjtot' != 'bklx[112 chars]pwmw,sllbie,trieh,vpyho,wetox,wntnomxeyv,wuolruphy,yjfpj,zjtot'\n",
      "- bklxgytfa,eftitc,gccwaqss,gjfkz,gnasdovglw,hcvsgl,hvzyqvwurd,jiry,kbllbxaplh,lhnd,lotlj,mqalbblp,nbjdopwol,onlhvlfg,pwmw,sllbie,trieh,wetox,wntnomxeyv,wuolruphy,yjfpj,zjtot\n",
      "+ bklxgytfa,eftitc,gccwaqss,gjfkz,gnasdovglw,hcvsgl,hvzyqvwurd,jiry,kbllbxaplh,lhnd,lotlj,mqalbblp,nbjdopwol,onlhvlfg,pwmw,sllbie,trieh,vpyho,wetox,wntnomxeyv,wuolruphy,yjfpj,zjtot\n",
      "?                                                                                                                                       ++++++\n",
      " : Should be bklxgytfa,eftitc,gccwaqss,gjfkz,gnasdovglw,hcvsgl,hvzyqvwurd,jiry,kbllbxaplh,lhnd,lotlj,mqalbblp,nbjdopwol,onlhvlfg,pwmw,sllbie,trieh,vpyho,wetox,wntnomxeyv,wuolruphy,yjfpj,zjtot\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'awocov,iobfixblr,jbejnc,kbvexvume,kvdoszhd,nqiqs[50 chars]aeii' != 'awocov,gvufk,iobfixblr,jbejnc,kbvexvume,kvdoszhd[56 chars]aeii'\n",
      "- awocov,iobfixblr,jbejnc,kbvexvume,kvdoszhd,nqiqs,ohfkxfkc,pnqwepkw,pntiymhzfv,tcdhy,vuqsenw,xpofpnaeii\n",
      "+ awocov,gvufk,iobfixblr,jbejnc,kbvexvume,kvdoszhd,nqiqs,ohfkxfkc,pnqwepkw,pntiymhzfv,tcdhy,vuqsenw,xpofpnaeii\n",
      "?       ++++++\n",
      " : Should be awocov,gvufk,iobfixblr,jbejnc,kbvexvume,kvdoszhd,nqiqs,ohfkxfkc,pnqwepkw,pntiymhzfv,tcdhy,vuqsenw,xpofpnaeii\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'ntdqegzj,riqhvwkd,udjtiy,wgnv,wiucrrjnl,wiyc,xgzvdt,yamcbl' != 'gdmzjzym,ntdqegzj,riqhvwkd,udjtiy,wgnv,wiucrrjnl,wiyc,xgzvdt,yamcbl'\n",
      "- ntdqegzj,riqhvwkd,udjtiy,wgnv,wiucrrjnl,wiyc,xgzvdt,yamcbl\n",
      "+ gdmzjzym,ntdqegzj,riqhvwkd,udjtiy,wgnv,wiucrrjnl,wiyc,xgzvdt,yamcbl\n",
      "? +++++++++\n",
      " : Should be gdmzjzym,ntdqegzj,riqhvwkd,udjtiy,wgnv,wiucrrjnl,wiyc,xgzvdt,yamcbl\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'csqt[54 chars]yncbn,kvcefavqa,qzylakobme,rmrsm,soispjbqqx,sqqes,ysyiief' != 'csqt[54 chars]yncbn,kvcefavqa,lyrv,qzylakobme,rmrsm,soispjbqqx,sqqes,ysyiief'\n",
      "- csqtpx,fgjedgmdj,ggnsae,gqqumossah,iljynrz,kaugxjijop,kaviyncbn,kvcefavqa,qzylakobme,rmrsm,soispjbqqx,sqqes,ysyiief\n",
      "+ csqtpx,fgjedgmdj,ggnsae,gqqumossah,iljynrz,kaugxjijop,kaviyncbn,kvcefavqa,lyrv,qzylakobme,rmrsm,soispjbqqx,sqqes,ysyiief\n",
      "?                                                                           +++++\n",
      " : Should be csqtpx,fgjedgmdj,ggnsae,gqqumossah,iljynrz,kaugxjijop,kaviyncbn,kvcefavqa,lyrv,qzylakobme,rmrsm,soispjbqqx,sqqes,ysyiief\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'ehfkmko,kmafod,lyxgmnxd' != 'ehfkmko,kmafod,lyxgmnxd,vtqdxuho'\n",
      "- ehfkmko,kmafod,lyxgmnxd\n",
      "+ ehfkmko,kmafod,lyxgmnxd,vtqdxuho\n",
      "?                        +++++++++\n",
      " : Should be ehfkmko,kmafod,lyxgmnxd,vtqdxuho\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'ewap[23 chars]lzlk,icvonuzf,lrztb,mhycipfgq,mkfwpdj,myeclamq[63 chars]efdb' != 'ewap[23 chars]lzlk,gytfrpw,icvonuzf,lrztb,mhycipfgq,mkfwpdj,[71 chars]efdb'\n",
      "- ewapjri,fmcdk,fzhhvnqufz,gslzlk,icvonuzf,lrztb,mhycipfgq,mkfwpdj,myeclamqnh,oqyogqqw,qhrkgw,qlssea,teklkgnux,ucne,wnicxcp,wvzjttjyi,zkegefdb\n",
      "+ ewapjri,fmcdk,fzhhvnqufz,gslzlk,gytfrpw,icvonuzf,lrztb,mhycipfgq,mkfwpdj,myeclamqnh,oqyogqqw,qhrkgw,qlssea,teklkgnux,ucne,wnicxcp,wvzjttjyi,zkegefdb\n",
      "?                                ++++++++\n",
      " : Should be ewapjri,fmcdk,fzhhvnqufz,gslzlk,gytfrpw,icvonuzf,lrztb,mhycipfgq,mkfwpdj,myeclamqnh,oqyogqqw,qhrkgw,qlssea,teklkgnux,ucne,wnicxcp,wvzjttjyi,zkegefdb\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'amwx[25 chars]ayyau,jisqdm,jmccyeks,jqiv,tgnhfcqxqy,wjcibr,xwxnvph' != 'amwx[25 chars]ayyau,jisqdm,jmccyeks,jqiv,rsrpkdncc,tgnhfcqxqy,wjcibr,xwxnvph'\n",
      "- amwxntgsx,cnsz,dqouxnb,drqj,fayyau,jisqdm,jmccyeks,jqiv,tgnhfcqxqy,wjcibr,xwxnvph\n",
      "+ amwxntgsx,cnsz,dqouxnb,drqj,fayyau,jisqdm,jmccyeks,jqiv,rsrpkdncc,tgnhfcqxqy,wjcibr,xwxnvph\n",
      "?                                                         ++++++++++\n",
      " : Should be amwxntgsx,cnsz,dqouxnb,drqj,fayyau,jisqdm,jmccyeks,jqiv,rsrpkdncc,tgnhfcqxqy,wjcibr,xwxnvph\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'ctln[20 chars]rgrj,lnmty,nmrdwxga,rmfumppx,scobhdw,tbwykm,tz[30 chars]wrcz' != 'ctln[20 chars]rgrj,knbgq,lnmty,nmrdwxga,rmfumppx,scobhdw,tbw[36 chars]wrcz'\n",
      "- ctlnofuf,ezqwxggmdq,fvgdrgrj,lnmty,nmrdwxga,rmfumppx,scobhdw,tbwykm,tzfpqnoh,ujjai,uqtdicbbqz,vzzln,wrcz\n",
      "+ ctlnofuf,ezqwxggmdq,fvgdrgrj,knbgq,lnmty,nmrdwxga,rmfumppx,scobhdw,tbwykm,tzfpqnoh,ujjai,uqtdicbbqz,vzzln,wrcz\n",
      "?                             ++++++\n",
      " : Should be ctlnofuf,ezqwxggmdq,fvgdrgrj,knbgq,lnmty,nmrdwxga,rmfumppx,scobhdw,tbwykm,tzfpqnoh,ujjai,uqtdicbbqz,vzzln,wrcz\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bxpr[78 chars]mnatjdafz,oojlf,oyzp,qepssvsey,vuzfzngl,wtmumxo,yoimkryvm' != 'bxpr[78 chars]mnatjdafz,oojlf,oyzp,qepssvsey,vuzfzngl,wtmumxo,ymdg,yoimkryvm'\n",
      "- bxprga,cgtzebxlb,emeegun,fgka,groiuhh,hntoqynvbb,jfgqxbm,keozbhiy,lmtk,lvdwpkgvjv,mnatjdafz,oojlf,oyzp,qepssvsey,vuzfzngl,wtmumxo,yoimkryvm\n",
      "+ bxprga,cgtzebxlb,emeegun,fgka,groiuhh,hntoqynvbb,jfgqxbm,keozbhiy,lmtk,lvdwpkgvjv,mnatjdafz,oojlf,oyzp,qepssvsey,vuzfzngl,wtmumxo,ymdg,yoimkryvm\n",
      "?                                                                                                                                    +++++\n",
      " : Should be bxprga,cgtzebxlb,emeegun,fgka,groiuhh,hntoqynvbb,jfgqxbm,keozbhiy,lmtk,lvdwpkgvjv,mnatjdafz,oojlf,oyzp,qepssvsey,vuzfzngl,wtmumxo,ymdg,yoimkryvm\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'djyv[22 chars]jnatnji,qzip,scxcntrmhw,ubyealghra,wjvcuxgag,xnnqswnr' != 'djyv[22 chars]jnatnji,qzip,scxcntrmhw,ubyealghra,vvgxqqzb,wjvcuxgag,xnnqswnr'\n",
      "- djyvlmnl,fvpyzcjux,haaqfq,jnatnji,qzip,scxcntrmhw,ubyealghra,wjvcuxgag,xnnqswnr\n",
      "+ djyvlmnl,fvpyzcjux,haaqfq,jnatnji,qzip,scxcntrmhw,ubyealghra,vvgxqqzb,wjvcuxgag,xnnqswnr\n",
      "?                                                              +++++++++\n",
      " : Should be djyvlmnl,fvpyzcjux,haaqfq,jnatnji,qzip,scxcntrmhw,ubyealghra,vvgxqqzb,wjvcuxgag,xnnqswnr\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'ccjj[20 chars]qoy,lzievsqv,mpacxhiioy,omaj,rcduobjaw,rlgq,tk[37 chars]ghht' != 'ccjj[20 chars]qoy,ldsj,lzievsqv,mpacxhiioy,omaj,rcduobjaw,rl[42 chars]ghht'\n",
      "- ccjjdc,gjxzejrfa,jmwzbsrqoy,lzievsqv,mpacxhiioy,omaj,rcduobjaw,rlgq,tkcb,vjdesh,xdako,yoylxhe,zdjjmseibg,zsghht\n",
      "+ ccjjdc,gjxzejrfa,jmwzbsrqoy,ldsj,lzievsqv,mpacxhiioy,omaj,rcduobjaw,rlgq,tkcb,vjdesh,xdako,yoylxhe,zdjjmseibg,zsghht\n",
      "?                            +++++\n",
      " : Should be ccjjdc,gjxzejrfa,jmwzbsrqoy,ldsj,lzievsqv,mpacxhiioy,omaj,rcduobjaw,rlgq,tkcb,vjdesh,xdako,yoylxhe,zdjjmseibg,zsghht\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'brljggmvnc,brwvrmpijk,ephlcu,jbfeqpn,jdozuwjgyj,mxhi' != 'brljggmvnc,brwvrmpijk,ephlcu,jbfeqpn,jdozuwjgyj,mxhi,sngum'\n",
      "- brljggmvnc,brwvrmpijk,ephlcu,jbfeqpn,jdozuwjgyj,mxhi\n",
      "+ brljggmvnc,brwvrmpijk,ephlcu,jbfeqpn,jdozuwjgyj,mxhi,sngum\n",
      "?                                                     ++++++\n",
      " : Should be brljggmvnc,brwvrmpijk,ephlcu,jbfeqpn,jdozuwjgyj,mxhi,sngum\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bxrx[60 chars]m,mtszxgl,odvbzfg,oplyjrk,vzto,xjvxqs,ygrkyhuh,zzszlautc' != 'bxrx[60 chars]m,mtszxgl,odvbzfg,oplyjrk,psuif,vzto,xjvxqs,ygrkyhuh,zzszlautc'\n",
      "- bxrxy,eahjw,fcunx,gwhyx,hblieoun,hlzgyzaw,ieckvabux,joobhwe,jtfum,mtszxgl,odvbzfg,oplyjrk,vzto,xjvxqs,ygrkyhuh,zzszlautc\n",
      "+ bxrxy,eahjw,fcunx,gwhyx,hblieoun,hlzgyzaw,ieckvabux,joobhwe,jtfum,mtszxgl,odvbzfg,oplyjrk,psuif,vzto,xjvxqs,ygrkyhuh,zzszlautc\n",
      "?                                                                                           ++++++\n",
      " : Should be bxrxy,eahjw,fcunx,gwhyx,hblieoun,hlzgyzaw,ieckvabux,joobhwe,jtfum,mtszxgl,odvbzfg,oplyjrk,psuif,vzto,xjvxqs,ygrkyhuh,zzszlautc\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'aiso[63 chars]gc,nxkcydxqbh,otdytuqrfo,pjlsxdffqx,qrtcpyhlf,sxcbjpkck' != 'aiso[63 chars]gc,nxkcydxqbh,otdytuqrfo,pjlsxdffqx,qrtcpyhlf,rvpiqa,sxcbjpkck'\n",
      "- aisotfe,cgaxxozgzr,cjgxgwv,cwspuboc,dbworj,gwsnnkau,jxpqgtgw,nalfrygc,nxkcydxqbh,otdytuqrfo,pjlsxdffqx,qrtcpyhlf,sxcbjpkck\n",
      "+ aisotfe,cgaxxozgzr,cjgxgwv,cwspuboc,dbworj,gwsnnkau,jxpqgtgw,nalfrygc,nxkcydxqbh,otdytuqrfo,pjlsxdffqx,qrtcpyhlf,rvpiqa,sxcbjpkck\n",
      "?                                                                                                                  +++++++\n",
      " : Should be aisotfe,cgaxxozgzr,cjgxgwv,cwspuboc,dbworj,gwsnnkau,jxpqgtgw,nalfrygc,nxkcydxqbh,otdytuqrfo,pjlsxdffqx,qrtcpyhlf,rvpiqa,sxcbjpkck\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'atfu[49 chars]mmaz,pccdbcdvrw,pwidmylm,scshrl,sgroxyfl,tmjur[28 chars]xzxj' != 'atfu[49 chars]mmaz,npdbvyfmy,pccdbcdvrw,pwidmylm,scshrl,sgro[38 chars]xzxj'\n",
      "- atfuy,bawflcvq,gcuepgw,hxtuudhp,iytxuqtnfn,jrbakdfbr,mmaz,pccdbcdvrw,pwidmylm,scshrl,sgroxyfl,tmjurokq,tpaw,tyhlnute,ugnnnldvx,xzxj\n",
      "+ atfuy,bawflcvq,gcuepgw,hxtuudhp,iytxuqtnfn,jrbakdfbr,mmaz,npdbvyfmy,pccdbcdvrw,pwidmylm,scshrl,sgroxyfl,tmjurokq,tpaw,tyhlnute,ugnnnldvx,xzxj\n",
      "?                                                          ++++++++++\n",
      " : Should be atfuy,bawflcvq,gcuepgw,hxtuudhp,iytxuqtnfn,jrbakdfbr,mmaz,npdbvyfmy,pccdbcdvrw,pwidmylm,scshrl,sgroxyfl,tmjurokq,tpaw,tyhlnute,ugnnnldvx,xzxj\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bnwb[43 chars]gkbg,oggsorywql,oibxllv,pibecdnv,puningsdxc,qf[88 chars]mjtw' != 'bnwb[43 chars]gkbg,ltgmdfjw,oggsorywql,oibxllv,pibecdnv,puni[97 chars]mjtw'\n",
      "- bnwbd,ctrnlx,dauivciao,dpbrtcy,eizpqhbcfp,etvb,gkbg,oggsorywql,oibxllv,pibecdnv,puningsdxc,qfpkvkww,rnhhmosmlx,shpspoz,slamnyvq,szhmlgkmas,toangp,uazzlbl,upurzgwpw,vcjzey,votuxr,xcpmjtw\n",
      "+ bnwbd,ctrnlx,dauivciao,dpbrtcy,eizpqhbcfp,etvb,gkbg,ltgmdfjw,oggsorywql,oibxllv,pibecdnv,puningsdxc,qfpkvkww,rnhhmosmlx,shpspoz,slamnyvq,szhmlgkmas,toangp,uazzlbl,upurzgwpw,vcjzey,votuxr,xcpmjtw\n",
      "?                                                    +++++++++\n",
      " : Should be bnwbd,ctrnlx,dauivciao,dpbrtcy,eizpqhbcfp,etvb,gkbg,ltgmdfjw,oggsorywql,oibxllv,pibecdnv,puningsdxc,qfpkvkww,rnhhmosmlx,shpspoz,slamnyvq,szhmlgkmas,toangp,uazzlbl,upurzgwpw,vcjzey,votuxr,xcpmjtw\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'ainc[35 chars]b,lhehou,pgozdtya,pjxbu,rgty,tkqktcg,twrhlltlc,xswngj' != 'ainc[35 chars]b,lhehou,mvclrqun,pgozdtya,pjxbu,rgty,tkqktcg,twrhlltlc,xswngj'\n",
      "- ainctpj,azrvyhbybm,epahyxijf,fbtil,iuunb,lhehou,pgozdtya,pjxbu,rgty,tkqktcg,twrhlltlc,xswngj\n",
      "+ ainctpj,azrvyhbybm,epahyxijf,fbtil,iuunb,lhehou,mvclrqun,pgozdtya,pjxbu,rgty,tkqktcg,twrhlltlc,xswngj\n",
      "?                                                 +++++++++\n",
      " : Should be ainctpj,azrvyhbybm,epahyxijf,fbtil,iuunb,lhehou,mvclrqun,pgozdtya,pjxbu,rgty,tkqktcg,twrhlltlc,xswngj\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bswb[99 chars]byspzq,pqyiqy,qewbnq,qimyue,swoxeor,tvdhrox,xdyigfvyxc' != 'bswb[99 chars]byspzq,pqyiqy,qewbnq,qimyue,swoxeor,tvdhrox,vjgtcdk,xdyigfvyxc'\n",
      "- bswbuueai,bwey,cpqg,dgtzzfxji,duzwzlcu,foneuh,fqynripsqs,icmzdb,jnqvapwfdv,korm,myuzoxol,nwnf,ofmimub,pbyspzq,pqyiqy,qewbnq,qimyue,swoxeor,tvdhrox,xdyigfvyxc\n",
      "+ bswbuueai,bwey,cpqg,dgtzzfxji,duzwzlcu,foneuh,fqynripsqs,icmzdb,jnqvapwfdv,korm,myuzoxol,nwnf,ofmimub,pbyspzq,pqyiqy,qewbnq,qimyue,swoxeor,tvdhrox,vjgtcdk,xdyigfvyxc\n",
      "?                                                                                                                                                    ++++++++\n",
      " : Should be bswbuueai,bwey,cpqg,dgtzzfxji,duzwzlcu,foneuh,fqynripsqs,icmzdb,jnqvapwfdv,korm,myuzoxol,nwnf,ofmimub,pbyspzq,pqyiqy,qewbnq,qimyue,swoxeor,tvdhrox,vjgtcdk,xdyigfvyxc\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'dqookpmjuv,jdvkcrnx,jnauzuaatl,mztpwzz,qew[50 chars]xwvk' != 'dlzu,dqookpmjuv,jdvkcrnx,jnauzuaatl,mztpwz[55 chars]xwvk'\n",
      "- dqookpmjuv,jdvkcrnx,jnauzuaatl,mztpwzz,qewmfors,qkjwrjo,rlfiurqj,snmxcmfzml,tauok,tzsulab,xuxwvk\n",
      "+ dlzu,dqookpmjuv,jdvkcrnx,jnauzuaatl,mztpwzz,qewmfors,qkjwrjo,rlfiurqj,snmxcmfzml,tauok,tzsulab,xuxwvk\n",
      "? +++++\n",
      " : Should be dlzu,dqookpmjuv,jdvkcrnx,jnauzuaatl,mztpwzz,qewmfors,qkjwrjo,rlfiurqj,snmxcmfzml,tauok,tzsulab,xuxwvk\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'eamh[56 chars]tpk,pnpbhdgvvh,pvyssk,rmulerka,sgkokvyhsf,shhc[58 chars]ylxh' != 'eamh[56 chars]tpk,pcugvttsj,pnpbhdgvvh,pvyssk,rmulerka,sgkok[68 chars]ylxh'\n",
      "- eamhhom,hquxw,imkumih,jlpexiyr,jvlgrxjbh,lkacyg,mewtsce,nowntpk,pnpbhdgvvh,pvyssk,rmulerka,sgkokvyhsf,shhcwpx,umuxjyib,wapdlv,xcwcluodsp,xdddmjs,yfpjvfpyi,yimcbfxc,ylxh\n",
      "+ eamhhom,hquxw,imkumih,jlpexiyr,jvlgrxjbh,lkacyg,mewtsce,nowntpk,pcugvttsj,pnpbhdgvvh,pvyssk,rmulerka,sgkokvyhsf,shhcwpx,umuxjyib,wapdlv,xcwcluodsp,xdddmjs,yfpjvfpyi,yimcbfxc,ylxh\n",
      "?                                                                ++++++++++\n",
      " : Should be eamhhom,hquxw,imkumih,jlpexiyr,jvlgrxjbh,lkacyg,mewtsce,nowntpk,pcugvttsj,pnpbhdgvvh,pvyssk,rmulerka,sgkokvyhsf,shhcwpx,umuxjyib,wapdlv,xcwcluodsp,xdddmjs,yfpjvfpyi,yimcbfxc,ylxh\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'aeuvd,bnmgn,eehj,erpdbtoxy,esrjfs,kmghl,oyhrxbd,phwcy[79 chars]mbku' != 'aeuvd,bnmgn,bvfvd,eehj,erpdbtoxy,esrjfs,kmghl,oyhrxbd[85 chars]mbku'\n",
      "- aeuvd,bnmgn,eehj,erpdbtoxy,esrjfs,kmghl,oyhrxbd,phwcyhxt,pxehagcenq,qobdonjpk,qytmx,smjpfrziix,uuwecpgxb,wwulx,xuotdnf,zdghcbzp,zlutmbku\n",
      "+ aeuvd,bnmgn,bvfvd,eehj,erpdbtoxy,esrjfs,kmghl,oyhrxbd,phwcyhxt,pxehagcenq,qobdonjpk,qytmx,smjpfrziix,uuwecpgxb,wwulx,xuotdnf,zdghcbzp,zlutmbku\n",
      "?            ++++++\n",
      " : Should be aeuvd,bnmgn,bvfvd,eehj,erpdbtoxy,esrjfs,kmghl,oyhrxbd,phwcyhxt,pxehagcenq,qobdonjpk,qytmx,smjpfrziix,uuwecpgxb,wwulx,xuotdnf,zdghcbzp,zlutmbku\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bkxd[17 chars]drxt,gejjjvw,ggla,hjqpe,kxkkd,laitqeufks,onpdn[61 chars]srgh' != 'bkxd[17 chars]drxt,epqcczmjvj,gejjjvw,ggla,hjqpe,kxkkd,laitq[72 chars]srgh'\n",
      "- bkxdfa,bvhgx,ceuxjau,drxt,gejjjvw,ggla,hjqpe,kxkkd,laitqeufks,onpdn,qcfnclmx,rqrmn,tgckmheuyy,uonf,vuoa,wrwoxrcg,xaalsqiyps,ymzdsrgh\n",
      "+ bkxdfa,bvhgx,ceuxjau,drxt,epqcczmjvj,gejjjvw,ggla,hjqpe,kxkkd,laitqeufks,onpdn,qcfnclmx,rqrmn,tgckmheuyy,uonf,vuoa,wrwoxrcg,xaalsqiyps,ymzdsrgh\n",
      "?                          +++++++++++\n",
      " : Should be bkxdfa,bvhgx,ceuxjau,drxt,epqcczmjvj,gejjjvw,ggla,hjqpe,kxkkd,laitqeufks,onpdn,qcfnclmx,rqrmn,tgckmheuyy,uonf,vuoa,wrwoxrcg,xaalsqiyps,ymzdsrgh\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'djwl[50 chars]te,msibxei,natmbu,pvlnqbtka,unnlshlk,xmtnerhza,zbzbnakp' != 'djwl[50 chars]te,msibxei,natmbu,pvlnqbtka,unnlshlk,wnwtqj,xmtnerhza,zbzbnakp'\n",
      "- djwllrmlv,gjoyj,hrpjezuhou,iyqkr,ldkfs,lhyhufg,mfqpkxrte,msibxei,natmbu,pvlnqbtka,unnlshlk,xmtnerhza,zbzbnakp\n",
      "+ djwllrmlv,gjoyj,hrpjezuhou,iyqkr,ldkfs,lhyhufg,mfqpkxrte,msibxei,natmbu,pvlnqbtka,unnlshlk,wnwtqj,xmtnerhza,zbzbnakp\n",
      "?                                                                                            +++++++\n",
      " : Should be djwllrmlv,gjoyj,hrpjezuhou,iyqkr,ldkfs,lhyhufg,mfqpkxrte,msibxei,natmbu,pvlnqbtka,unnlshlk,wnwtqj,xmtnerhza,zbzbnakp\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'cegb[30 chars]iaef,jyfnpgzqsk,mdersbslr,mmngiqyf,pskovmmlxb,[17 chars]xlbd' != 'cegb[30 chars]iaef,gzfwvb,jyfnpgzqsk,mdersbslr,mmngiqyf,psko[24 chars]xlbd'\n",
      "- cegbujtxnr,dlbztsxm,ezixhnwp,fijhxiaef,jyfnpgzqsk,mdersbslr,mmngiqyf,pskovmmlxb,rnxf,tdspry,tlnd,xlbd\n",
      "+ cegbujtxnr,dlbztsxm,ezixhnwp,fijhxiaef,gzfwvb,jyfnpgzqsk,mdersbslr,mmngiqyf,pskovmmlxb,rnxf,tdspry,tlnd,xlbd\n",
      "?                                       +++++++\n",
      " : Should be cegbujtxnr,dlbztsxm,ezixhnwp,fijhxiaef,gzfwvb,jyfnpgzqsk,mdersbslr,mmngiqyf,pskovmmlxb,rnxf,tdspry,tlnd,xlbd\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'dqrq[20 chars],iovpf,iplko,jgroc,kjhwjafp,lfco,mwipr,tfciqaognm,wqzy' != 'dqrq[20 chars],iovpf,iplko,jgroc,kjhwjafp,lfco,mwipr,tfciqaognm,wqzy,yzmhihs'\n",
      "- dqrqsm,ejuednahni,ezsaij,iovpf,iplko,jgroc,kjhwjafp,lfco,mwipr,tfciqaognm,wqzy\n",
      "+ dqrqsm,ejuednahni,ezsaij,iovpf,iplko,jgroc,kjhwjafp,lfco,mwipr,tfciqaognm,wqzy,yzmhihs\n",
      "?                                                                               ++++++++\n",
      " : Should be dqrqsm,ejuednahni,ezsaij,iovpf,iplko,jgroc,kjhwjafp,lfco,mwipr,tfciqaognm,wqzy,yzmhihs\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'sagyqbneb,yeqjnkw,zqtdi' != 'hwdypcd,sagyqbneb,yeqjnkw,zqtdi'\n",
      "- sagyqbneb,yeqjnkw,zqtdi\n",
      "+ hwdypcd,sagyqbneb,yeqjnkw,zqtdi\n",
      "? ++++++++\n",
      " : Should be hwdypcd,sagyqbneb,yeqjnkw,zqtdi\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'askg[72 chars]rpxg,pnwmcfo,sxellotz,tblb,tydaa,umcxg,vkidico[66 chars]befu' != 'askg[72 chars]rpxg,ozywrltwde,pnwmcfo,sxellotz,tblb,tydaa,um[77 chars]befu'\n",
      "- askgvyw,ddlaqxpp,fsenrqh,hdckukopq,hjbq,jsbwsxrxlu,lmtubcpyl,mtbo,mvfe,ngkgirpxg,pnwmcfo,sxellotz,tblb,tydaa,umcxg,vkidicos,wamwyrmaq,wbcidc,wikheichf,wjhbqo,wjykdcl,wswsjjinx,wurmjra,ytyybefu\n",
      "?                                                                                  ^^^^^^^^^^^^^^^\n",
      "+ askgvyw,ddlaqxpp,fsenrqh,hdckukopq,hjbq,jsbwsxrxlu,lmtubcpyl,mtbo,mvfe,ngkgirpxg,ozywrltwde,pnwmcfo,sxellotz,tblb,tydaa,umcxg,vkidicos,wamwyrmaq,wbcidc,wikheichf,wjhbqo,wjykdcl,wswsjjinx,wurmjra,ytyybefu\n",
      "?                                                                                  ^ +++++++++++++++++++++++++\n",
      " : Should be askgvyw,ddlaqxpp,fsenrqh,hdckukopq,hjbq,jsbwsxrxlu,lmtubcpyl,mtbo,mvfe,ngkgirpxg,ozywrltwde,pnwmcfo,sxellotz,tblb,tydaa,umcxg,vkidicos,wamwyrmaq,wbcidc,wikheichf,wjhbqo,wjykdcl,wswsjjinx,wurmjra,ytyybefu\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bqyq[102 chars],ndtluperlr,ozueujzlvo,qdpvb,ubqfb,voqzmj,wkqwjwdagb,yaam' != 'bqyq[102 chars],ndtluperlr,ozueujzlvo,qdpvb,ubqfb,vjwm,voqzmj,wkqwjwdagb,yaam'\n",
      "- bqyqytv,cuazgcuga,fahk,fasoppd,fhjwv,gaisszldny,gqdyr,grxtzgvi,hdqs,hlsg,hugmi,jrttlv,mgsfffxsp,mlqwelyjri,ndtluperlr,ozueujzlvo,qdpvb,ubqfb,voqzmj,wkqwjwdagb,yaam\n",
      "+ bqyqytv,cuazgcuga,fahk,fasoppd,fhjwv,gaisszldny,gqdyr,grxtzgvi,hdqs,hlsg,hugmi,jrttlv,mgsfffxsp,mlqwelyjri,ndtluperlr,ozueujzlvo,qdpvb,ubqfb,vjwm,voqzmj,wkqwjwdagb,yaam\n",
      "?                                                                                                                                               +++++\n",
      " : Should be bqyqytv,cuazgcuga,fahk,fasoppd,fhjwv,gaisszldny,gqdyr,grxtzgvi,hdqs,hlsg,hugmi,jrttlv,mgsfffxsp,mlqwelyjri,ndtluperlr,ozueujzlvo,qdpvb,ubqfb,vjwm,voqzmj,wkqwjwdagb,yaam\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'eafmyz,fdifyvdfy,fpml,hqktegsgwz,kjgldvrl[138 chars]yxas' != 'dkafh,eafmyz,fdifyvdfy,fpml,hqktegsgwz,kj[144 chars]yxas'\n",
      "- eafmyz,fdifyvdfy,fpml,hqktegsgwz,kjgldvrljs,lfie,lweef,msuah,mtzqdiy,nida,onxcbx,pohyicolg,rbuxnqjui,siritottv,soutbuz,tjayx,tssjdtmwou,tyju,umzkpx,urrghkiv,vmnxeyfjx,wbhtkvvon,zcyxas\n",
      "+ dkafh,eafmyz,fdifyvdfy,fpml,hqktegsgwz,kjgldvrljs,lfie,lweef,msuah,mtzqdiy,nida,onxcbx,pohyicolg,rbuxnqjui,siritottv,soutbuz,tjayx,tssjdtmwou,tyju,umzkpx,urrghkiv,vmnxeyfjx,wbhtkvvon,zcyxas\n",
      "? ++++++\n",
      " : Should be dkafh,eafmyz,fdifyvdfy,fpml,hqktegsgwz,kjgldvrljs,lfie,lweef,msuah,mtzqdiy,nida,onxcbx,pohyicolg,rbuxnqjui,siritottv,soutbuz,tjayx,tssjdtmwou,tyju,umzkpx,urrghkiv,vmnxeyfjx,wbhtkvvon,zcyxas\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'nhvhukhhnf,qndafvty,slygbkzzfo' != 'dwkch,nhvhukhhnf,qndafvty,slygbkzzfo'\n",
      "- nhvhukhhnf,qndafvty,slygbkzzfo\n",
      "+ dwkch,nhvhukhhnf,qndafvty,slygbkzzfo\n",
      "? ++++++\n",
      " : Should be dwkch,nhvhukhhnf,qndafvty,slygbkzzfo\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'cfwzpvdhlt,chhqmvmq,hendmrjthe,hetotlbk,jbfewkese,jwlyyfomnn,[45 chars]qtbf' != 'cfwzpvdhlt,chhqmvmq,djnj,hendmrjthe,hetotlbk,jbfewkese,jwlyyf[50 chars]qtbf'\n",
      "- cfwzpvdhlt,chhqmvmq,hendmrjthe,hetotlbk,jbfewkese,jwlyyfomnn,kotnvwa,lktviqtpe,oqhtk,tsgg,wjggrxu,wqbcsd,zqtbf\n",
      "+ cfwzpvdhlt,chhqmvmq,djnj,hendmrjthe,hetotlbk,jbfewkese,jwlyyfomnn,kotnvwa,lktviqtpe,oqhtk,tsgg,wjggrxu,wqbcsd,zqtbf\n",
      "?                    +++++\n",
      " : Should be cfwzpvdhlt,chhqmvmq,djnj,hendmrjthe,hetotlbk,jbfewkese,jwlyyfomnn,kotnvwa,lktviqtpe,oqhtk,tsgg,wjggrxu,wqbcsd,zqtbf\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'ajda[101 chars]pr,twthvfhh,tzkaxn,urzorrzf,wklam,xeahgf,yxzvqreh,zzxpcfi' != 'ajda[101 chars]pr,twthvfhh,tzkaxn,urzorrzf,wklam,xeahgf,yxzvqreh,zaps,zzxpcfi'\n",
      "- ajdalngrlq,azirk,biznu,dzcwjgo,fldkclrx,gpifcyda,gvegsly,iqqsfu,kfjgaw,mprmb,pntsvceggs,scovi,sgnn,shhzykpr,twthvfhh,tzkaxn,urzorrzf,wklam,xeahgf,yxzvqreh,zzxpcfi\n",
      "+ ajdalngrlq,azirk,biznu,dzcwjgo,fldkclrx,gpifcyda,gvegsly,iqqsfu,kfjgaw,mprmb,pntsvceggs,scovi,sgnn,shhzykpr,twthvfhh,tzkaxn,urzorrzf,wklam,xeahgf,yxzvqreh,zaps,zzxpcfi\n",
      "?                                                                                                                                                             +++++\n",
      " : Should be ajdalngrlq,azirk,biznu,dzcwjgo,fldkclrx,gpifcyda,gvegsly,iqqsfu,kfjgaw,mprmb,pntsvceggs,scovi,sgnn,shhzykpr,twthvfhh,tzkaxn,urzorrzf,wklam,xeahgf,yxzvqreh,zaps,zzxpcfi\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'awlq[52 chars]wsxwju,hslgnhood,ijkafaxe,lkzzbce,ppxjlfgzj,tbcbnchb' != 'awlq[52 chars]wsxwju,hslgnhood,ijkafaxe,lkzzbce,ppxjlfgzj,tbcbnchb,ygzprljyc'\n",
      "- awlqr,azjutp,cmgu,damnavr,emkyjwltye,enlal,esvxjhaow,fecwsxwju,hslgnhood,ijkafaxe,lkzzbce,ppxjlfgzj,tbcbnchb\n",
      "+ awlqr,azjutp,cmgu,damnavr,emkyjwltye,enlal,esvxjhaow,fecwsxwju,hslgnhood,ijkafaxe,lkzzbce,ppxjlfgzj,tbcbnchb,ygzprljyc\n",
      "?                                                                                                             ++++++++++\n",
      " : Should be awlqr,azjutp,cmgu,damnavr,emkyjwltye,enlal,esvxjhaow,fecwsxwju,hslgnhood,ijkafaxe,lkzzbce,ppxjlfgzj,tbcbnchb,ygzprljyc\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bdfm[109 chars]vdu,rqcqfw,rzancp,sxnp,vcjazey,voomfenxay,xndo,yhfvudoqb' != 'bdfm[109 chars]vdu,rqcqfw,rxobm,rzancp,sxnp,vcjazey,voomfenxay,xndo,yhfvudoqb'\n",
      "- bdfmyj,bemftyb,fithms,fiyfzeyyn,gonzvl,ijnmixookn,iwrroehe,jwkr,liooc,lkkxy,mbicwm,nezbkeu,oftcrbuce,paifwobre,revdu,rqcqfw,rzancp,sxnp,vcjazey,voomfenxay,xndo,yhfvudoqb\n",
      "+ bdfmyj,bemftyb,fithms,fiyfzeyyn,gonzvl,ijnmixookn,iwrroehe,jwkr,liooc,lkkxy,mbicwm,nezbkeu,oftcrbuce,paifwobre,revdu,rqcqfw,rxobm,rzancp,sxnp,vcjazey,voomfenxay,xndo,yhfvudoqb\n",
      "?                                                                                                                              ++++++\n",
      " : Should be bdfmyj,bemftyb,fithms,fiyfzeyyn,gonzvl,ijnmixookn,iwrroehe,jwkr,liooc,lkkxy,mbicwm,nezbkeu,oftcrbuce,paifwobre,revdu,rqcqfw,rxobm,rzancp,sxnp,vcjazey,voomfenxay,xndo,yhfvudoqb\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'aeruu,azkfo,crjsyv,dperf,fkeqpmela,gbwlv,ittkp,jadxyx,jajj,jm[51 chars]xdrm' != 'aeruu,azkfo,crjsyv,dmmccpnc,dperf,fkeqpmela,gbwlv,ittkp,jadxy[60 chars]xdrm'\n",
      "- aeruu,azkfo,crjsyv,dperf,fkeqpmela,gbwlv,ittkp,jadxyx,jajj,jmqdgbx,neuts,pknaovk,qxebjpcti,vnbzpwryby,wwcog,zqvxxdrm\n",
      "+ aeruu,azkfo,crjsyv,dmmccpnc,dperf,fkeqpmela,gbwlv,ittkp,jadxyx,jajj,jmqdgbx,neuts,pknaovk,qxebjpcti,vnbzpwryby,wwcog,zqvxxdrm\n",
      "?                   +++++++++\n",
      " : Should be aeruu,azkfo,crjsyv,dmmccpnc,dperf,fkeqpmela,gbwlv,ittkp,jadxyx,jajj,jmqdgbx,neuts,pknaovk,qxebjpcti,vnbzpwryby,wwcog,zqvxxdrm\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'dnajat,jdhtfkge,jshzssv,tbcpofjvkr,tllkb,vwlu,zzctmfchgk' != 'dhcvabhbpv,dnajat,jdhtfkge,jshzssv,tbcpofjvkr,tllkb,vwlu,zzctmfchgk'\n",
      "- dnajat,jdhtfkge,jshzssv,tbcpofjvkr,tllkb,vwlu,zzctmfchgk\n",
      "+ dhcvabhbpv,dnajat,jdhtfkge,jshzssv,tbcpofjvkr,tllkb,vwlu,zzctmfchgk\n",
      "? +++++++++++\n",
      " : Should be dhcvabhbpv,dnajat,jdhtfkge,jshzssv,tbcpofjvkr,tllkb,vwlu,zzctmfchgk\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'aahu[56 chars]jhor,jfhxsv,jllcodvx,jszbiel,qhrbc,qxpa,rmudw,[18 chars]zixt' != 'aahu[56 chars]jhor,iiah,jfhxsv,jllcodvx,jszbiel,qhrbc,qxpa,r[23 chars]zixt'\n",
      "- aahurt,alcjzqbmcb,bbloxrncwi,cxnjvwyad,efdcc,etxe,fgvgqvm,fvjhor,jfhxsv,jllcodvx,jszbiel,qhrbc,qxpa,rmudw,sknygeg,vybkhubpn,zixt\n",
      "+ aahurt,alcjzqbmcb,bbloxrncwi,cxnjvwyad,efdcc,etxe,fgvgqvm,fvjhor,iiah,jfhxsv,jllcodvx,jszbiel,qhrbc,qxpa,rmudw,sknygeg,vybkhubpn,zixt\n",
      "?                                                                  +++++\n",
      " : Should be aahurt,alcjzqbmcb,bbloxrncwi,cxnjvwyad,efdcc,etxe,fgvgqvm,fvjhor,iiah,jfhxsv,jllcodvx,jszbiel,qhrbc,qxpa,rmudw,sknygeg,vybkhubpn,zixt\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'jfynbxjkxw,mxcyt,nrly,onstr' != 'jfynbxjkxw,mxcyt,nrly,onstr,xkfitoqnlu'\n",
      "- jfynbxjkxw,mxcyt,nrly,onstr\n",
      "+ jfynbxjkxw,mxcyt,nrly,onstr,xkfitoqnlu\n",
      "?                            +++++++++++\n",
      " : Should be jfynbxjkxw,mxcyt,nrly,onstr,xkfitoqnlu\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bllx[99 chars]dlz,rpoezb,sxaikhbs,tfgb,xdmec,xxinswiwxx,yqvg,zbeafvtswz' != 'bllx[99 chars]dlz,rpoezb,sxaikhbs,tfgb,xdmec,xxinswiwxx,yqvg,zbeafvtswz,znor'\n",
      "- bllxjgw,cemua,crqtyn,cwhpsf,dyvvtk,ixtkqnd,jyczwbgkh,luzfscu,npqmtwg,oakrmvm,osseqvyl,rgvzmuoen,rmfdsqgdlz,rpoezb,sxaikhbs,tfgb,xdmec,xxinswiwxx,yqvg,zbeafvtswz\n",
      "+ bllxjgw,cemua,crqtyn,cwhpsf,dyvvtk,ixtkqnd,jyczwbgkh,luzfscu,npqmtwg,oakrmvm,osseqvyl,rgvzmuoen,rmfdsqgdlz,rpoezb,sxaikhbs,tfgb,xdmec,xxinswiwxx,yqvg,zbeafvtswz,znor\n",
      "?                                                                                                                                                                 +++++\n",
      " : Should be bllxjgw,cemua,crqtyn,cwhpsf,dyvvtk,ixtkqnd,jyczwbgkh,luzfscu,npqmtwg,oakrmvm,osseqvyl,rgvzmuoen,rmfdsqgdlz,rpoezb,sxaikhbs,tfgb,xdmec,xxinswiwxx,yqvg,zbeafvtswz,znor\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'ahzw[29 chars]tkh,evsyyqs,hfaap,hmewvu,iqus,ogrfyfdffh,peruu[34 chars]yrod' != 'ahzw[29 chars]tkh,eunhymvtzv,evsyyqs,hfaap,hmewvu,iqus,ogrfy[45 chars]yrod'\n",
      "- ahzwlbdtti,akzpsmlon,bhnxusifjs,etkh,evsyyqs,hfaap,hmewvu,iqus,ogrfyfdffh,peruuv,tcwb,uhhdbgjo,xcqthpdapk,yewhhx,yrod\n",
      "+ ahzwlbdtti,akzpsmlon,bhnxusifjs,etkh,eunhymvtzv,evsyyqs,hfaap,hmewvu,iqus,ogrfyfdffh,peruuv,tcwb,uhhdbgjo,xcqthpdapk,yewhhx,yrod\n",
      "?                                     +++++++++++\n",
      " : Should be ahzwlbdtti,akzpsmlon,bhnxusifjs,etkh,eunhymvtzv,evsyyqs,hfaap,hmewvu,iqus,ogrfyfdffh,peruuv,tcwb,uhhdbgjo,xcqthpdapk,yewhhx,yrod\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'amox[35 chars]bsnrl,jgnjmv,qjln,qzih,tips,ueeebeyh,umgijaoq,zipcssfesn' != 'amox[35 chars]bsnrl,jgnjmv,qjln,qzih,tips,ueeebeyh,umgijaoq,ymdyx,zipcssfesn'\n",
      "- amoxjmp,dcdlm,fckgt,frsygps,fteuimje,gebsnrl,jgnjmv,qjln,qzih,tips,ueeebeyh,umgijaoq,zipcssfesn\n",
      "+ amoxjmp,dcdlm,fckgt,frsygps,fteuimje,gebsnrl,jgnjmv,qjln,qzih,tips,ueeebeyh,umgijaoq,ymdyx,zipcssfesn\n",
      "?                                                                                      ++++++\n",
      " : Should be amoxjmp,dcdlm,fckgt,frsygps,fteuimje,gebsnrl,jgnjmv,qjln,qzih,tips,ueeebeyh,umgijaoq,ymdyx,zipcssfesn\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'cecson,juufzvd,lkgk,xftsezxiu,yphb' != 'cecson,juufzvd,lkgk,xftsezxiu,yphb,zauqfa'\n",
      "- cecson,juufzvd,lkgk,xftsezxiu,yphb\n",
      "+ cecson,juufzvd,lkgk,xftsezxiu,yphb,zauqfa\n",
      "?                                   +++++++\n",
      " : Should be cecson,juufzvd,lkgk,xftsezxiu,yphb,zauqfa\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'ataywiku,egdkjtp,enxksjbtm,hrhqg,jcphacfs,jplrczwsfd,jzes,n[29 chars]kdbc' != 'ataywiku,egdkjtp,ehucv,enxksjbtm,hrhqg,jcphacfs,jplrczwsfd,[35 chars]kdbc'\n",
      "- ataywiku,egdkjtp,enxksjbtm,hrhqg,jcphacfs,jplrczwsfd,jzes,ntxc,ocdsrqysh,ocutctp,ubpdf,wkdbc\n",
      "+ ataywiku,egdkjtp,ehucv,enxksjbtm,hrhqg,jcphacfs,jplrczwsfd,jzes,ntxc,ocdsrqysh,ocutctp,ubpdf,wkdbc\n",
      "?                 ++++++\n",
      " : Should be ataywiku,egdkjtp,ehucv,enxksjbtm,hrhqg,jcphacfs,jplrczwsfd,jzes,ntxc,ocdsrqysh,ocutctp,ubpdf,wkdbc\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bjfr[136 chars]nk,usksrkvnef,vexxynxw,vneg,xesbrivyc,xrpnp,ybwqgpikw' != 'bjfr[136 chars]nk,usksrkvnef,vexxynxw,vneg,xesbrivyc,xrpnp,ybwqgpikw,yetsvmlh'\n",
      "- bjfrlfcqiw,bvzbjotvz,dtggvf,ejdcjq,evkfdbv,fceksc,gmkvxjyha,hgzsteti,kgppddjkn,lcwiw,letjz,mnhlqlsvj,mwfdzoi,pavtxfcnzv,qzpp,sysa,umixib,uosnk,usksrkvnef,vexxynxw,vneg,xesbrivyc,xrpnp,ybwqgpikw\n",
      "+ bjfrlfcqiw,bvzbjotvz,dtggvf,ejdcjq,evkfdbv,fceksc,gmkvxjyha,hgzsteti,kgppddjkn,lcwiw,letjz,mnhlqlsvj,mwfdzoi,pavtxfcnzv,qzpp,sysa,umixib,uosnk,usksrkvnef,vexxynxw,vneg,xesbrivyc,xrpnp,ybwqgpikw,yetsvmlh\n",
      "?                                                                                                                                                                                                  +++++++++\n",
      " : Should be bjfrlfcqiw,bvzbjotvz,dtggvf,ejdcjq,evkfdbv,fceksc,gmkvxjyha,hgzsteti,kgppddjkn,lcwiw,letjz,mnhlqlsvj,mwfdzoi,pavtxfcnzv,qzpp,sysa,umixib,uosnk,usksrkvnef,vexxynxw,vneg,xesbrivyc,xrpnp,ybwqgpikw,yetsvmlh\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'byxr[66 chars]crg,jvwelnthln,ksiytwkf,mjieovio,ngnfhjtaw,sxiwl,xaxkxwq' != 'byxr[66 chars]crg,jvwelnthln,ksiytwkf,mjieovio,ngnfhjtaw,roiwv,sxiwl,xaxkxwq'\n",
      "- byxrrdr,cnvox,eveashpt,eyzdhxb,hmhc,htcagmpuex,hvfoqxxzs,jeiviuma,jueqcrg,jvwelnthln,ksiytwkf,mjieovio,ngnfhjtaw,sxiwl,xaxkxwq\n",
      "+ byxrrdr,cnvox,eveashpt,eyzdhxb,hmhc,htcagmpuex,hvfoqxxzs,jeiviuma,jueqcrg,jvwelnthln,ksiytwkf,mjieovio,ngnfhjtaw,roiwv,sxiwl,xaxkxwq\n",
      "?                                                                                                                  ++++++\n",
      " : Should be byxrrdr,cnvox,eveashpt,eyzdhxb,hmhc,htcagmpuex,hvfoqxxzs,jeiviuma,jueqcrg,jvwelnthln,ksiytwkf,mjieovio,ngnfhjtaw,roiwv,sxiwl,xaxkxwq\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bwrc,cnene,epgye,ftclvclnoq,hqmtypsmmk,idepseeq,jnyw[87 chars]zxpc' != 'bwrc,cnene,cznbr,epgye,ftclvclnoq,hqmtypsmmk,idepsee[93 chars]zxpc'\n",
      "- bwrc,cnene,epgye,ftclvclnoq,hqmtypsmmk,idepseeq,jnywl,lbzygn,lkltmrrl,mrug,oenefx,ofiidsobe,pmsx,usykxepi,uufx,wnrjwhon,wunipyuuh,xyxykael,zxpc\n",
      "+ bwrc,cnene,cznbr,epgye,ftclvclnoq,hqmtypsmmk,idepseeq,jnywl,lbzygn,lkltmrrl,mrug,oenefx,ofiidsobe,pmsx,usykxepi,uufx,wnrjwhon,wunipyuuh,xyxykael,zxpc\n",
      "?           ++++++\n",
      " : Should be bwrc,cnene,cznbr,epgye,ftclvclnoq,hqmtypsmmk,idepseeq,jnywl,lbzygn,lkltmrrl,mrug,oenefx,ofiidsobe,pmsx,usykxepi,uufx,wnrjwhon,wunipyuuh,xyxykael,zxpc\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'arhm[28 chars]glzz,feyvsb,hdgwr,kowhb,kpdxwvss,kuiwfrbx,kzeh[91 chars]gdsc' != 'arhm[28 chars]glzz,engxwfwy,feyvsb,hdgwr,kowhb,kpdxwvss,kuiw[100 chars]gdsc'\n",
      "- arhmpgs,creopoq,dcrhvf,deemx,dtvglzz,feyvsb,hdgwr,kowhb,kpdxwvss,kuiwfrbx,kzehn,niihsrk,nxojm,qgvagw,qwjuai,ramiotc,rsbwzwpzj,rxuyyb,snrukprgr,tiue,usaxucfm,zfwtsgt,zrwrgdsc\n",
      "+ arhmpgs,creopoq,dcrhvf,deemx,dtvglzz,engxwfwy,feyvsb,hdgwr,kowhb,kpdxwvss,kuiwfrbx,kzehn,niihsrk,nxojm,qgvagw,qwjuai,ramiotc,rsbwzwpzj,rxuyyb,snrukprgr,tiue,usaxucfm,zfwtsgt,zrwrgdsc\n",
      "?                                     +++++++++\n",
      " : Should be arhmpgs,creopoq,dcrhvf,deemx,dtvglzz,engxwfwy,feyvsb,hdgwr,kowhb,kpdxwvss,kuiwfrbx,kzehn,niihsrk,nxojm,qgvagw,qwjuai,ramiotc,rsbwzwpzj,rxuyyb,snrukprgr,tiue,usaxucfm,zfwtsgt,zrwrgdsc\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'biso[31 chars],rijayrlfkz,rnrzxfye,skukcfflm,tyws,uupgmsyc,zzdfdlhew' != 'biso[31 chars],rijayrlfkz,rnrzxfye,rspbkbv,skukcfflm,tyws,uupgmsyc,zzdfdlhew'\n",
      "- bisoi,crsehbwhkz,jxwbjrvik,pfvrylwp,rijayrlfkz,rnrzxfye,skukcfflm,tyws,uupgmsyc,zzdfdlhew\n",
      "+ bisoi,crsehbwhkz,jxwbjrvik,pfvrylwp,rijayrlfkz,rnrzxfye,rspbkbv,skukcfflm,tyws,uupgmsyc,zzdfdlhew\n",
      "?                                                         ++++++++\n",
      " : Should be bisoi,crsehbwhkz,jxwbjrvik,pfvrylwp,rijayrlfkz,rnrzxfye,rspbkbv,skukcfflm,tyws,uupgmsyc,zzdfdlhew\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'brkk[86 chars]gmn,kqhwuw,lwhkg,mjgofqweku,nglkxufc,rvwhfeqlz[57 chars]kzmy' != 'brkk[86 chars]gmn,kphb,kqhwuw,lwhkg,mjgofqweku,nglkxufc,rvwh[62 chars]kzmy'\n",
      "- brkkgamkhc,caujpsc,cbrv,efxrxzzz,flhzersm,frxcn,ixffyeff,jazikdsoyi,jvdozddq,kfayiu,kfrfbhgmn,kqhwuw,lwhkg,mjgofqweku,nglkxufc,rvwhfeqlz,sjdwetqkh,tinxmrkdcy,tujavh,uqbi,vkfzn,vysuiwz,wthf,xvyskzmy\n",
      "+ brkkgamkhc,caujpsc,cbrv,efxrxzzz,flhzersm,frxcn,ixffyeff,jazikdsoyi,jvdozddq,kfayiu,kfrfbhgmn,kphb,kqhwuw,lwhkg,mjgofqweku,nglkxufc,rvwhfeqlz,sjdwetqkh,tinxmrkdcy,tujavh,uqbi,vkfzn,vysuiwz,wthf,xvyskzmy\n",
      "?                                                                                                +++++\n",
      " : Should be brkkgamkhc,caujpsc,cbrv,efxrxzzz,flhzersm,frxcn,ixffyeff,jazikdsoyi,jvdozddq,kfayiu,kfrfbhgmn,kphb,kqhwuw,lwhkg,mjgofqweku,nglkxufc,rvwhfeqlz,sjdwetqkh,tinxmrkdcy,tujavh,uqbi,vkfzn,vysuiwz,wthf,xvyskzmy\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'cesf[87 chars]oln,plrrk,unvttw,uvwrdowcy,vbzz,xusryhejhx,ybr[21 chars]hgkd' != 'cesf[87 chars]oln,pflafb,plrrk,unvttw,uvwrdowcy,vbzz,xusryhe[28 chars]hgkd'\n",
      "- cesfstpe,citaxjytk,famaeqvksc,fkblzej,guuyylu,hlchipnfc,hxtoal,kbcmssmlm,mzkr,nbdvgk,nplaunoln,plrrk,unvttw,uvwrdowcy,vbzz,xusryhejhx,ybrrtjgft,zgdfssyqh,zjgkhgkd\n",
      "+ cesfstpe,citaxjytk,famaeqvksc,fkblzej,guuyylu,hlchipnfc,hxtoal,kbcmssmlm,mzkr,nbdvgk,nplaunoln,pflafb,plrrk,unvttw,uvwrdowcy,vbzz,xusryhejhx,ybrrtjgft,zgdfssyqh,zjgkhgkd\n",
      "?                                                                                                 +++++++\n",
      " : Should be cesfstpe,citaxjytk,famaeqvksc,fkblzej,guuyylu,hlchipnfc,hxtoal,kbcmssmlm,mzkr,nbdvgk,nplaunoln,pflafb,plrrk,unvttw,uvwrdowcy,vbzz,xusryhejhx,ybrrtjgft,zgdfssyqh,zjgkhgkd\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bycl[73 chars]puugnqc,qqawrmnj,rlhkamw,sedpw,sutoquhzeb,txjqgar,xoltlx' != 'bycl[73 chars]puugnqc,qqawrmnj,rlhkamw,sedpw,sutoquhzeb,txjqgar,xoltlx,zdsvs'\n",
      "- byclngmbz,donfrev,dzbjvluro,eiowe,fsewzxqwmp,gmniyw,hbuom,ijbmgva,incvrdvdzi,puugnqc,qqawrmnj,rlhkamw,sedpw,sutoquhzeb,txjqgar,xoltlx\n",
      "+ byclngmbz,donfrev,dzbjvluro,eiowe,fsewzxqwmp,gmniyw,hbuom,ijbmgva,incvrdvdzi,puugnqc,qqawrmnj,rlhkamw,sedpw,sutoquhzeb,txjqgar,xoltlx,zdsvs\n",
      "?                                                                                                                                      ++++++\n",
      " : Should be byclngmbz,donfrev,dzbjvluro,eiowe,fsewzxqwmp,gmniyw,hbuom,ijbmgva,incvrdvdzi,puugnqc,qqawrmnj,rlhkamw,sedpw,sutoquhzeb,txjqgar,xoltlx,zdsvs\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'dwdc[68 chars]tgzt,qcnozc,qfio,rrbydczlud,sabbmjiep,sgawkk,twcztreg,zstbmej' != 'dwdc[68 chars]tgzt,pcrj,qcnozc,qfio,rrbydczlud,sabbmjiep,sga[16 chars]bmej'\n",
      "- dwdcuhw,fczllmi,fkmvaij,hbvkwkerqx,igvfdksris,ktexahi,nwjfjb,olxxf,ovlictgzt,qcnozc,qfio,rrbydczlud,sabbmjiep,sgawkk,twcztreg,zstbmej\n",
      "+ dwdcuhw,fczllmi,fkmvaij,hbvkwkerqx,igvfdksris,ktexahi,nwjfjb,olxxf,ovlictgzt,pcrj,qcnozc,qfio,rrbydczlud,sabbmjiep,sgawkk,twcztreg,zstbmej\n",
      "?                                                                              +++++\n",
      " : Should be dwdcuhw,fczllmi,fkmvaij,hbvkwkerqx,igvfdksris,ktexahi,nwjfjb,olxxf,ovlictgzt,pcrj,qcnozc,qfio,rrbydczlud,sabbmjiep,sgawkk,twcztreg,zstbmej\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'alss[61 chars]cnzb,lktfpum,mqlpjwsk,ngksfy,owedn,qrecfw,qvhr[39 chars]hown' != 'alss[61 chars]cnzb,jxajfh,lktfpum,mqlpjwsk,ngksfy,owedn,qrec[46 chars]hown'\n",
      "- alssesqlv,ccqf,dlyi,drenmb,ejutydjun,fncayp,gwrcgct,hlprqjwgw,istcnzb,lktfpum,mqlpjwsk,ngksfy,owedn,qrecfw,qvhr,qyvg,umpgfkh,vytblzridn,ybjm,yjsw,yocfhown\n",
      "+ alssesqlv,ccqf,dlyi,drenmb,ejutydjun,fncayp,gwrcgct,hlprqjwgw,istcnzb,jxajfh,lktfpum,mqlpjwsk,ngksfy,owedn,qrecfw,qvhr,qyvg,umpgfkh,vytblzridn,ybjm,yjsw,yocfhown\n",
      "?                                                                      +++++++\n",
      " : Should be alssesqlv,ccqf,dlyi,drenmb,ejutydjun,fncayp,gwrcgct,hlprqjwgw,istcnzb,jxajfh,lktfpum,mqlpjwsk,ngksfy,owedn,qrecfw,qvhr,qyvg,umpgfkh,vytblzridn,ybjm,yjsw,yocfhown\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'aanr[74 chars]h,kgxgjfezh,mrnmmsohs,mvqgpgpujt,oobshd,sojx,zkpcfedjm' != 'aanr[74 chars]h,kgxgjfezh,mrnmmsohs,mvqgpgpujt,oobshd,sojx,spzwvdc,zkpcfedjm'\n",
      "- aanrhh,abtyffhd,aioecbvq,cdawi,fgrjcwsck,fzuy,fzwd,gmscltz,hrhk,hufy,iuljpgvnhh,kgxgjfezh,mrnmmsohs,mvqgpgpujt,oobshd,sojx,zkpcfedjm\n",
      "+ aanrhh,abtyffhd,aioecbvq,cdawi,fgrjcwsck,fzuy,fzwd,gmscltz,hrhk,hufy,iuljpgvnhh,kgxgjfezh,mrnmmsohs,mvqgpgpujt,oobshd,sojx,spzwvdc,zkpcfedjm\n",
      "?                                                                                                                            ++++++++\n",
      " : Should be aanrhh,abtyffhd,aioecbvq,cdawi,fgrjcwsck,fzuy,fzwd,gmscltz,hrhk,hufy,iuljpgvnhh,kgxgjfezh,mrnmmsohs,mvqgpgpujt,oobshd,sojx,spzwvdc,zkpcfedjm\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bkac[84 chars]ddzm,oflon,qdbrrwnz,rsqbgt,tnxp,ulibewf,zhhhte,zhtk' != 'bkac[84 chars]ddzm,oflon,qdbrrwnz,rsqbgt,tnxp,tqcmmzekdx,ulibewf,zhhhte,zhtk'\n",
      "- bkact,cisec,djdpajmnk,dnuykil,ishohusre,kboez,kjfuwbxn,lyblouh,mbqaxm,mksbn,mpxrhts,nrolddzm,oflon,qdbrrwnz,rsqbgt,tnxp,ulibewf,zhhhte,zhtk\n",
      "+ bkact,cisec,djdpajmnk,dnuykil,ishohusre,kboez,kjfuwbxn,lyblouh,mbqaxm,mksbn,mpxrhts,nrolddzm,oflon,qdbrrwnz,rsqbgt,tnxp,tqcmmzekdx,ulibewf,zhhhte,zhtk\n",
      "?                                                                                                                         +++++++++++\n",
      " : Should be bkact,cisec,djdpajmnk,dnuykil,ishohusre,kboez,kjfuwbxn,lyblouh,mbqaxm,mksbn,mpxrhts,nrolddzm,oflon,qdbrrwnz,rsqbgt,tnxp,tqcmmzekdx,ulibewf,zhhhte,zhtk\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'adyx[13 chars]fsyv,gqkvgeyys,gsid,ijogurtf,jicik,kbzgekdpx,p[13 chars]vktd' != 'adyx[13 chars]fsyv,fxhpiowco,gqkvgeyys,gsid,ijogurtf,jicik,k[23 chars]vktd'\n",
      "- adyxagb,bhpke,fnkfsyv,gqkvgeyys,gsid,ijogurtf,jicik,kbzgekdpx,pzum,qiwju,xgevktd\n",
      "+ adyxagb,bhpke,fnkfsyv,fxhpiowco,gqkvgeyys,gsid,ijogurtf,jicik,kbzgekdpx,pzum,qiwju,xgevktd\n",
      "?                      ++++++++++\n",
      " : Should be adyxagb,bhpke,fnkfsyv,fxhpiowco,gqkvgeyys,gsid,ijogurtf,jicik,kbzgekdpx,pzum,qiwju,xgevktd\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'eocw[49 chars]vcfg,lokxrdayg,lshajzb,mierzjaypz,nohrl,qbnufv[89 chars]ysoa' != 'eocw[49 chars]vcfg,kzrmv,lokxrdayg,lshajzb,mierzjaypz,nohrl,[95 chars]ysoa'\n",
      "- eocwwfd,eqgvzah,ernnzugva,fsblmhz,hzudqlm,jelxj,kiqkcvcfg,lokxrdayg,lshajzb,mierzjaypz,nohrl,qbnufvuzu,qbwu,rasdiuama,rcvyaurjgr,rgic,rgljqhytnb,spchmbu,tzwivam,vpmmbm,vrycqwvkc,wzktgb,zugysoa\n",
      "+ eocwwfd,eqgvzah,ernnzugva,fsblmhz,hzudqlm,jelxj,kiqkcvcfg,kzrmv,lokxrdayg,lshajzb,mierzjaypz,nohrl,qbnufvuzu,qbwu,rasdiuama,rcvyaurjgr,rgic,rgljqhytnb,spchmbu,tzwivam,vpmmbm,vrycqwvkc,wzktgb,zugysoa\n",
      "?                                                          ++++++\n",
      " : Should be eocwwfd,eqgvzah,ernnzugva,fsblmhz,hzudqlm,jelxj,kiqkcvcfg,kzrmv,lokxrdayg,lshajzb,mierzjaypz,nohrl,qbnufvuzu,qbwu,rasdiuama,rcvyaurjgr,rgic,rgljqhytnb,spchmbu,tzwivam,vpmmbm,vrycqwvkc,wzktgb,zugysoa\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'drzz[33 chars],ksmz,kysh,lrtyoruvl,ssomr,teizlx,tzptg,vykix,zacrdnqcfd' != 'drzz[33 chars],ksmz,kysh,lrtyoruvl,ssomr,teizlx,tzptg,vykix,yzzwp,zacrdnqcfd'\n",
      "- drzzxd,itcd,jooqbthwob,jzwgacuq,kehvk,ksmz,kysh,lrtyoruvl,ssomr,teizlx,tzptg,vykix,zacrdnqcfd\n",
      "+ drzzxd,itcd,jooqbthwob,jzwgacuq,kehvk,ksmz,kysh,lrtyoruvl,ssomr,teizlx,tzptg,vykix,yzzwp,zacrdnqcfd\n",
      "?                                                                                    ++++++\n",
      " : Should be drzzxd,itcd,jooqbthwob,jzwgacuq,kehvk,ksmz,kysh,lrtyoruvl,ssomr,teizlx,tzptg,vykix,yzzwp,zacrdnqcfd\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'detznd,mfowmcsxf,udud' != 'detznd,mfowmcsxf,oczxvmdpyr,udud'\n",
      "- detznd,mfowmcsxf,udud\n",
      "+ detznd,mfowmcsxf,oczxvmdpyr,udud\n",
      "?                  +++++++++++\n",
      " : Should be detznd,mfowmcsxf,oczxvmdpyr,udud\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'bpiz[76 chars]paes,ndsjpcgy,odyswauwu,pktsqzsar,tvlnchdy,usii,ylbsw,yzimfe' != 'bpiz[76 chars]paes,lxspayhfv,ndsjpcgy,odyswauwu,pktsqzsar,tv[20 chars]imfe'\n",
      "- bpiztpaoji,dopz,fich,frzp,ftemot,gtucou,ibdtspvewd,icwpplpm,kddl,lejfofmms,lsjrppaes,ndsjpcgy,odyswauwu,pktsqzsar,tvlnchdy,usii,ylbsw,yzimfe\n",
      "+ bpiztpaoji,dopz,fich,frzp,ftemot,gtucou,ibdtspvewd,icwpplpm,kddl,lejfofmms,lsjrppaes,lxspayhfv,ndsjpcgy,odyswauwu,pktsqzsar,tvlnchdy,usii,ylbsw,yzimfe\n",
      "?                                                                                      ++++++++++\n",
      " : Should be bpiztpaoji,dopz,fich,frzp,ftemot,gtucou,ibdtspvewd,icwpplpm,kddl,lejfofmms,lsjrppaes,lxspayhfv,ndsjpcgy,odyswauwu,pktsqzsar,tvlnchdy,usii,ylbsw,yzimfe\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'arkp[104 chars]iobhi,tiuoybnszu,tjotrhvrdz,tyqauq,uxqm,vmpoz,zvlrwu' != 'arkp[104 chars]iobhi,tiuoybnszu,tjotrhvrdz,tyqauq,uxqm,vmpoz,zvlrwu,zycnzxvqy'\n",
      "- arkprpnlh,buts,dcspwye,hkcpf,hxfg,idlcdmiwm,lfekpakml,lkbce,mddphnbq,nbutprkf,okrpwmt,osgv,qtvwmi,rcsth,rifyiobhi,tiuoybnszu,tjotrhvrdz,tyqauq,uxqm,vmpoz,zvlrwu\n",
      "+ arkprpnlh,buts,dcspwye,hkcpf,hxfg,idlcdmiwm,lfekpakml,lkbce,mddphnbq,nbutprkf,okrpwmt,osgv,qtvwmi,rcsth,rifyiobhi,tiuoybnszu,tjotrhvrdz,tyqauq,uxqm,vmpoz,zvlrwu,zycnzxvqy\n",
      "?                                                                                                                                                                 ++++++++++\n",
      " : Should be arkprpnlh,buts,dcspwye,hkcpf,hxfg,idlcdmiwm,lfekpakml,lkbce,mddphnbq,nbutprkf,okrpwmt,osgv,qtvwmi,rcsth,rifyiobhi,tiuoybnszu,tjotrhvrdz,tyqauq,uxqm,vmpoz,zvlrwu,zycnzxvqy\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'alwb,ekdzn,evgsrqoqv,faeuumug,fkvgo,iteythbtbh[142 chars]zdvw' != 'alwb,cpfbbnsmdv,ekdzn,evgsrqoqv,faeuumug,fkvgo[153 chars]zdvw'\n",
      "- alwb,ekdzn,evgsrqoqv,faeuumug,fkvgo,iteythbtbh,jgosmaoz,jzkfghvwa,kirkvzctct,mmhljvefda,mrhkkn,nmbags,peiqsse,qbxcvtbofb,qegdbf,rqrjov,usdm,vagmctxq,wtidxylltt,wzlilei,xgcy,ytczxn,yuyzghk,zdvw\n",
      "+ alwb,cpfbbnsmdv,ekdzn,evgsrqoqv,faeuumug,fkvgo,iteythbtbh,jgosmaoz,jzkfghvwa,kirkvzctct,mmhljvefda,mrhkkn,nmbags,peiqsse,qbxcvtbofb,qegdbf,rqrjov,usdm,vagmctxq,wtidxylltt,wzlilei,xgcy,ytczxn,yuyzghk,zdvw\n",
      " : Should be alwb,cpfbbnsmdv,ekdzn,evgsrqoqv,faeuumug,fkvgo,iteythbtbh,jgosmaoz,jzkfghvwa,kirkvzctct,mmhljvefda,mrhkkn,nmbags,peiqsse,qbxcvtbofb,qegdbf,rqrjov,usdm,vagmctxq,wtidxylltt,wzlilei,xgcy,ytczxn,yuyzghk,zdvw\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'efka,evbfhwgta,fomcizh,hsllpgp,mpyljbvhi,[27 chars]ukzp' != 'achxyudbf,efka,evbfhwgta,fomcizh,hsllpgp,[37 chars]ukzp'\n",
      "- efka,evbfhwgta,fomcizh,hsllpgp,mpyljbvhi,ogvdq,ovtbothpz,pkejhdtsog,ukzp\n",
      "+ achxyudbf,efka,evbfhwgta,fomcizh,hsllpgp,mpyljbvhi,ogvdq,ovtbothpz,pkejhdtsog,ukzp\n",
      "? ++++++++++\n",
      " : Should be achxyudbf,efka,evbfhwgta,fomcizh,hsllpgp,mpyljbvhi,ogvdq,ovtbothpz,pkejhdtsog,ukzp\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'aser[25 chars]lmke,eedj,ijubg,izbtk,kinsgnwdb,kwlwf,kwsuqvxy[109 chars]wxlt' != 'aser[25 chars]lmke,cupmq,eedj,ijubg,izbtk,kinsgnwdb,kwlwf,kw[115 chars]wxlt'\n",
      "- aserajboyr,bbfylbl,bojzxuf,btlmke,eedj,ijubg,izbtk,kinsgnwdb,kwlwf,kwsuqvxyp,lrmxidv,luxtv,nrbktzs,nxtzpdbysf,ocxfiuzlni,ogislaqhb,overdexty,phlmjbn,rmtonfhmp,vquo,xpiz,zacfw,zphheqm,zwxlt\n",
      "+ aserajboyr,bbfylbl,bojzxuf,btlmke,cupmq,eedj,ijubg,izbtk,kinsgnwdb,kwlwf,kwsuqvxyp,lrmxidv,luxtv,nrbktzs,nxtzpdbysf,ocxfiuzlni,ogislaqhb,overdexty,phlmjbn,rmtonfhmp,vquo,xpiz,zacfw,zphheqm,zwxlt\n",
      "?                                  ++++++\n",
      " : Should be aserajboyr,bbfylbl,bojzxuf,btlmke,cupmq,eedj,ijubg,izbtk,kinsgnwdb,kwlwf,kwsuqvxyp,lrmxidv,luxtv,nrbktzs,nxtzpdbysf,ocxfiuzlni,ogislaqhb,overdexty,phlmjbn,rmtonfhmp,vquo,xpiz,zacfw,zphheqm,zwxlt\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'feympwwun,sfatjsskc,szwvha' != 'feympwwun,gsbbrxtndz,sfatjsskc,szwvha'\n",
      "- feympwwun,sfatjsskc,szwvha\n",
      "+ feympwwun,gsbbrxtndz,sfatjsskc,szwvha\n",
      "?          +++++++++++\n",
      " : Should be feympwwun,gsbbrxtndz,sfatjsskc,szwvha\n",
      "\n",
      "======================================================================\n",
      "FAIL: runTest (mod.testing.test_alpha.<locals>.TestKnown)\n",
      "----------------------------------------------------------------------\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\annas\\Desktop\\lab-functions\\mod\\testing.py\", line 187, in runTest\n",
      "    self.assertEqual(fn(self.input), self.output, f\"Should be {self.output}\")\n",
      "AssertionError: 'cbqa[107 chars]j,qpkdth,rxityerl,vbsir,vpqfe,waphyisxyu,zhnaxe,zpso' != 'cbqa[107 chars]j,qpkdth,rxityerl,vbsir,vpqfe,waphyisxyu,wrlkjvsel,zhnaxe,zpso'\n",
      "- cbqaduwveq,ctonfshmkt,czdipgz,dgey,emhwplrs,fvukgaml,gsyuiq,heuo,jmzpf,kjgbqvuz,kzdm,mjpuv,mxhepx,odmmer,ozjejnj,qpkdth,rxityerl,vbsir,vpqfe,waphyisxyu,zhnaxe,zpso\n",
      "+ cbqaduwveq,ctonfshmkt,czdipgz,dgey,emhwplrs,fvukgaml,gsyuiq,heuo,jmzpf,kjgbqvuz,kzdm,mjpuv,mxhepx,odmmer,ozjejnj,qpkdth,rxityerl,vbsir,vpqfe,waphyisxyu,wrlkjvsel,zhnaxe,zpso\n",
      "?                                                                                                                                                         ++++++++++\n",
      " : Should be cbqaduwveq,ctonfshmkt,czdipgz,dgey,emhwplrs,fvukgaml,gsyuiq,heuo,jmzpf,kjgbqvuz,kzdm,mjpuv,mxhepx,odmmer,ozjejnj,qpkdth,rxityerl,vbsir,vpqfe,waphyisxyu,wrlkjvsel,zhnaxe,zpso\n",
      "\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.210s\n",
      "\n",
      "FAILED (failures=100)\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_alpha(sort_alpha)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 12. Write a function to check if a given password is strong (at least 8 characters, at least one lower case, at least one upper case, at least one number and at least one special character). It should output True if strong and False if not.\n",
    "`Valid special characters: # @ ! $ % & ( ) ^ * [ ] { }`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "## para comprobar la fortaleza de la contraseña se tienen que probar todas las condiciones\n",
    "## por tanto, se define un contador para cada una de las posibilidades\n",
    "def check_pass(string):\n",
    "    \"\"\"\n",
    "    Esta función toma un string dado y comprueba si se trata o no de una contraseña segura. \n",
    "    \\nPara ello, se iterará por cada uno de los caracteres y se contará cada y evaluará cada una de las letras, para ver si cumplen las condiciones necesarias. \n",
    "    arg:\n",
    "        string\n",
    "    return:\n",
    "        True --> la contraseña tiene al menos 8 caracteres, uno de tipo numérico, otro en mayúscula, otro en minúscula y otro de tipo especial\n",
    "        False --> se incumple al menos una de estas condiciones: menos 8 caracteres, uno de tipo numérico, otro en mayúscula, otro en minúscula y otro de tipo especial\n",
    "    \"\"\"\n",
    "    characters = 0 \n",
    "    lowercase = 0\n",
    "    uppercase = 0\n",
    "    number = 0\n",
    "    special_character = 0\n",
    "    special_character_lisr = ['#', '@', '!', '$', '%', '&', '(', ')', '^', '*', '[', ']', '{', '}']\n",
    "    for item in str(string):\n",
    "        #print(item)\n",
    "        characters += 1\n",
    "        if item.isnumeric():\n",
    "            number += 1\n",
    "        elif item.islower():\n",
    "            lowercase += 1\n",
    "        elif item.isupper():\n",
    "            uppercase += 1\n",
    "        elif item in special_character_lisr:\n",
    "            special_character += 1\n",
    "    if characters >= 8 and number>=1 and lowercase >= 1 and uppercase >= 1 and special_character >= 1:\n",
    "        #print(True, '-->' , characters, number, lowercase, uppercase, special_character)\n",
    "        return True\n",
    "    else:\n",
    "        #print(False, '-->' , characters, number, lowercase, uppercase, special_character)\n",
    "        return False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "....................................................................................................\n",
      "----------------------------------------------------------------------\n",
      "Ran 100 tests in 0.104s\n",
      "\n",
      "OK\n"
     ]
    }
   ],
   "source": [
    "# This will test your function \n",
    "test_pass(check_pass)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.9.13 ('iron')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  },
  "vscode": {
   "interpreter": {
    "hash": "e59d0b279e6461be5b4ea7b62838527cd8f41f3cbc9a1147b1a83b32d75ab689"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
